---
title: "Processing of aligned mIF ROIs"
author: "Nils Eling"
date: '`r Sys.Date()`'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

This script reads in channels of the acquired regions on the mIF data and
writes out multichannel images in `.tiff` format.

```{r}
library(EBImage)
library(tiff)

cur_folders <- list.files("~/SIB Swiss Institute of Bioinformatics/Robin Liechti - Workflow2021/Lausanne_data/ROI_registration_20220523/",
                          full.names = TRUE)

for (i in seq_along(cur_folders)) {
    if (length(list.files(cur_folders[i], pattern = "ROI_image.*.rds")) == 0) {
        next
    }
    
    if (!all.equal(list.files(cur_folders[i], pattern = "ROI_image.*.rds"),
                   c("ROI_image_autofluorescence.rds", "ROI_image_CD11c.rds", "ROI_image_CD15.rds", 
          "ROI_image_CD163.rds", "ROI_image_CD20.rds", "ROI_image_CD3.rds", 
          "ROI_image_CK.rds", "ROI_image_DAPI.rds"))) {
        stop("Wrong channel order")
    }
    
    cur_channels <- lapply(list.files(cur_folders[i], 
                                     pattern = "ROI_image.*.rds",
                                     full.names = TRUE),
                          function(x) {
                              return(readRDS(x))
                          })
    
    cur_channels <- abind(cur_channels, along = 3)
    
    writeImage(as.array(cur_channels)/(2^16 - 1), paste0("/Volumes/immucan_volume/processed_data/Panel_1/2022_WORKFLOW/mIF/steinbock/img/",
                                    basename(cur_folders[i]), ".tiff"),
               bits.per.sample = 16)
}

```