<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Comparison between IMC and IF</title>

<script src="site_libs/header-attrs-2.17/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">comparisons</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Comparison between IMC and IF</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-11-09
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>comparisons/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220429code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220429)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220429code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220429)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomImmucanWP7IMMUcanWorkflow2021tree9abd60d904088b36b1032d4a15f5df1c8e728081targetblank9abd60da">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/tree/9abd60d904088b36b1032d4a15f5df1c8e728081" target="_blank">9abd60d</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomImmucanWP7IMMUcanWorkflow2021tree9abd60d904088b36b1032d4a15f5df1c8e728081targetblank9abd60da"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/tree/9abd60d904088b36b1032d4a15f5df1c8e728081" target="_blank">9abd60d</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    Alignment/
    Ignored:    IMC/.DS_Store
    Ignored:    IMC/Analysis/.DS_Store
    Ignored:    IMC/Analysis/.Rproj.user/
    Ignored:    IMC/Analysis/docs/.DS_Store
    Ignored:    comparisons/.Rproj.user/
    Ignored:    comparisons/code/.ipynb_checkpoints/
    Ignored:    mIF/.DS_Store
    Ignored:    mIF/Analysis/.Rproj.user/

Unstaged changes:
    Modified:   comparisons/analysis/01_celltype_correlations.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>comparisons/analysis/01_celltype_correlations.Rmd</code>) and
HTML (<code>comparisons/docs/01_celltype_correlations.html</code>)
files. If you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/dc9748dd2e9cb938bfcc05e3a29bc20ec7a144df/comparisons/analysis/01_celltype_correlations.Rmd" target="_blank">dc9748d</a>
</td>
<td>
nilseling
</td>
<td>
2022-10-26
</td>
<td>
New comparisons
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/09d32f61a98b1b588620a3bd76317681b4a3743b/comparisons/analysis/01_celltype_correlations.Rmd" target="_blank">09d32f6</a>
</td>
<td>
nilseling
</td>
<td>
2022-06-30
</td>
<td>
Adjustments to cell type comparisons
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ImmucanWP7/IMMUcanWorkflow2021/09d32f61a98b1b588620a3bd76317681b4a3743b/comparisons/docs/01_celltype_correlations.html" target="_blank">09d32f6</a>
</td>
<td>
nilseling
</td>
<td>
2022-06-30
</td>
<td>
Adjustments to cell type comparisons
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/4554a259201ef6a52df476c8208dede506b41b04/comparisons/analysis/01_celltype_correlations.Rmd" target="_blank">4554a25</a>
</td>
<td>
nilseling
</td>
<td>
2022-06-24
</td>
<td>
Recompiled scripts
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ImmucanWP7/IMMUcanWorkflow2021/4554a259201ef6a52df476c8208dede506b41b04/comparisons/docs/01_celltype_correlations.html" target="_blank">4554a25</a>
</td>
<td>
nilseling
</td>
<td>
2022-06-24
</td>
<td>
Recompiled scripts
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/2a51289e9899968d8af665f68150b3cbf7eb3653/comparisons/analysis/01_celltype_correlations.Rmd" target="_blank">2a51289</a>
</td>
<td>
nilseling
</td>
<td>
2022-06-20
</td>
<td>
Read in data with local cooridnates
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/3a84dd86509fca85017d510c405b09733b2e3e26/comparisons/analysis/01_celltype_correlations.Rmd" target="_blank">3a84dd8</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-03
</td>
<td>
Adjusted base size
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ImmucanWP7/IMMUcanWorkflow2021/3a84dd86509fca85017d510c405b09733b2e3e26/comparisons/docs/01_celltype_correlations.html" target="_blank">3a84dd8</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-03
</td>
<td>
Adjusted base size
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/75d0a6df2a424569fc993d7f2034379ea37c1598/comparisons/analysis/01_celltype_correlations.Rmd" target="_blank">75d0a6d</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-02
</td>
<td>
Added differential abundance testing
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ImmucanWP7/IMMUcanWorkflow2021/75d0a6df2a424569fc993d7f2034379ea37c1598/comparisons/docs/01_celltype_correlations.html" target="_blank">75d0a6d</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-02
</td>
<td>
Added differential abundance testing
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/add09ed9175efe0986f12d7748eac4ca4d3fca7f/comparisons/analysis/01_celltype_correlations.Rmd" target="_blank">add09ed</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-01
</td>
<td>
Added marker expression comparison
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ImmucanWP7/IMMUcanWorkflow2021/add09ed9175efe0986f12d7748eac4ca4d3fca7f/comparisons/docs/01_celltype_correlations.html" target="_blank">add09ed</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-01
</td>
<td>
Added marker expression comparison
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/2a2141368265bcc975aeb5543b40fad8d4b93375/comparisons/analysis/01_celltype_correlations.Rmd" target="_blank">2a21413</a>
</td>
<td>
nilseling
</td>
<td>
2022-04-30
</td>
<td>
Celltype correlations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ImmucanWP7/IMMUcanWorkflow2021/2a2141368265bcc975aeb5543b40fad8d4b93375/comparisons/docs/01_celltype_correlations.html" target="_blank">2a21413</a>
</td>
<td>
nilseling
</td>
<td>
2022-04-30
</td>
<td>
Celltype correlations
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/c22d4e1fe74fdd577574bf3c0c514bb5df0a06fa/comparisons/analysis/01_celltype_correlations.Rmd" target="_blank">c22d4e1</a>
</td>
<td>
nilseling
</td>
<td>
2022-04-29
</td>
<td>
Added more projects
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>In this script, we will compare the celltypes observed in mIF with
those in IMC.</p>
<div id="load-data" class="section level2">
<h2>Load data</h2>
<p>First, we will read in the <code>SingleCellExperiment</code> objects
storing the IMC and IF data.</p>
<pre class="r"><code>library(SingleCellExperiment)
library(tidyverse)

sce_IMC &lt;- readRDS(&quot;/Volumes/immucan_volume/processed_data/Panel_1/2022_WORKFLOW/IMC/Rout/sce.rds&quot;)
sce_mIF &lt;- readRDS(&quot;/Volumes/immucan_volume/processed_data/Panel_1/2022_WORKFLOW/mIF/Rout/sce_ROIs.rds&quot;)

sce_IMC$matched_celltype &lt;- sce_IMC$external_matched_celltype

sce_IMC$matched_celltype[sce_IMC$matched_celltype == &quot;undefined&quot;] &lt;- &quot;other&quot;</code></pre>
</div>
<div id="number-of-cells" class="section level2">
<h2>Number of cells</h2>
<p>First, we will observe the number of cells per image.</p>
<pre class="r"><code>no_IMC &lt;- colData(sce_IMC) %&gt;% as_tibble() %&gt;%
  group_by(sample_id) %&gt;%
  summarize(count = n())
no_mIF &lt;- colData(sce_mIF) %&gt;% as_tibble() %&gt;%
  group_by(sample_id) %&gt;%
  summarize(count = n())

combined_df &lt;- merge(no_IMC, no_mIF, by=&quot;sample_id&quot;, sort=FALSE)
combined_df$patient_id &lt;- sub(&quot;-&quot;, &quot;&quot;, str_extract(combined_df$sample_id, &quot;[0-9]{8}-&quot;))

ggplot(combined_df) + 
  geom_point(aes(count.x, count.y, color = patient_id)) + 
  geom_abline(slope = 1, intercept = 0, color = &quot;dark red&quot;) +
  geom_smooth(aes(count.x, count.y), method=&#39;lm&#39;, formula= y~x, fullrange=TRUE) + 
  theme_minimal() + xlab(&quot;# Cells in IMC&quot;) + ylab(&quot;# Cells in mIF&quot;) +
  coord_fixed()</code></pre>
<p><img src="figure/01_celltype_correlations.Rmd/number-of-cells-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-number-of-cells-1">
Past versions of number-of-cells-1.png
</button>
</p>
<div id="fig-number-of-cells-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/4554a259201ef6a52df476c8208dede506b41b04/comparisons/docs/figure/01_celltype_correlations.Rmd/number-of-cells-1.png" target="_blank">4554a25</a>
</td>
<td>
nilseling
</td>
<td>
2022-06-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># With patient as covariate
ggplot(combined_df) + 
  geom_point(aes(count.x, count.y, color = patient_id)) + 
  geom_abline(slope = 1, intercept = 0, color = &quot;dark red&quot;) +
  geom_smooth(aes(count.x, count.y, color = patient_id), method=&#39;lm&#39;, formula= y~x,
              fullrange=FALSE, se= FALSE) + 
  theme_minimal() + xlab(&quot;# Cells in IMC&quot;) + ylab(&quot;# Cells in mIF&quot;) +
  coord_fixed()</code></pre>
<p><img src="figure/01_celltype_correlations.Rmd/number-of-cells-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-number-of-cells-2">
Past versions of number-of-cells-2.png
</button>
</p>
<div id="fig-number-of-cells-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/4554a259201ef6a52df476c8208dede506b41b04/comparisons/docs/figure/01_celltype_correlations.Rmd/number-of-cells-2.png" target="_blank">4554a25</a>
</td>
<td>
nilseling
</td>
<td>
2022-06-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cell-type-fractions" class="section level2">
<h2>Cell type fractions</h2>
<p>Next, we will compare the cell type fractions between the
technologies.</p>
<pre class="r"><code># Match cell types
cur_mIF &lt;- colData(sce_mIF)
cur_IMC &lt;- colData(sce_IMC)

frac_IMC &lt;- cur_IMC %&gt;% as_tibble() %&gt;%
  group_by(sample_id, matched_celltype, .drop = FALSE) %&gt;%
  summarize(count = n()) %&gt;%
  mutate(frac = count/sum(count))</code></pre>
<pre><code>`summarise()` has grouped output by &#39;sample_id&#39;. You can override using the
`.groups` argument.</code></pre>
<pre class="r"><code>frac_mIF &lt;- cur_mIF %&gt;% as_tibble() %&gt;%
  group_by(sample_id, matched_celltype, .drop = FALSE) %&gt;%
  summarize(count = n()) %&gt;%
  mutate(frac = count/sum(count))</code></pre>
<pre><code>`summarise()` has grouped output by &#39;sample_id&#39;. You can override using the
`.groups` argument.</code></pre>
<pre class="r"><code>combined_df &lt;- merge(frac_IMC, frac_mIF, by=c(&quot;sample_id&quot;, &quot;matched_celltype&quot;), sort=FALSE)
combined_df$patient_id &lt;- sub(&quot;-&quot;, &quot;&quot;, str_extract(combined_df$sample_id, &quot;[0-9]{8}-&quot;))

ggplot(combined_df) + 
  geom_point(aes(frac.x, frac.y, color = patient_id)) + 
  geom_abline(slope = 1, intercept = 0, color = &quot;dark red&quot;) +
  geom_smooth(aes(frac.x, frac.y), method=&#39;lm&#39;, formula= y~x, fullrange=TRUE) + 
  theme_classic(base_size = 20) + xlab(&quot;Fraction cell types in IMC&quot;) + ylab(&quot;Fraction cell types in mIF&quot;) +
  facet_wrap(.~matched_celltype, scales = &quot;fixed&quot;) </code></pre>
<p><img src="figure/01_celltype_correlations.Rmd/celltype-fractions-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-celltype-fractions-1">
Past versions of celltype-fractions-1.png
</button>
</p>
<div id="fig-celltype-fractions-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/4554a259201ef6a52df476c8208dede506b41b04/comparisons/docs/figure/01_celltype_correlations.Rmd/celltype-fractions-1.png" target="_blank">4554a25</a>
</td>
<td>
nilseling
</td>
<td>
2022-06-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/3a84dd86509fca85017d510c405b09733b2e3e26/comparisons/docs/figure/01_celltype_correlations.Rmd/celltype-fractions-1.png" target="_blank">3a84dd8</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/75d0a6df2a424569fc993d7f2034379ea37c1598/comparisons/docs/figure/01_celltype_correlations.Rmd/celltype-fractions-1.png" target="_blank">75d0a6d</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/2a2141368265bcc975aeb5543b40fad8d4b93375/comparisons/docs/figure/01_celltype_correlations.Rmd/celltype-fractions-1.png" target="_blank">2a21413</a>
</td>
<td>
nilseling
</td>
<td>
2022-04-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(combined_df) + 
  geom_smooth(aes(frac.x, frac.y, color = patient_id),  se = FALSE, method=&#39;lm&#39;, formula= y~x, fullrange=TRUE) + 
  geom_point(aes(frac.x, frac.y, color = patient_id)) + 
  geom_abline(slope = 1, intercept = 0, color = &quot;dark red&quot;) +
  theme_classic(base_size = 20) + xlab(&quot;Fraction cell types in IMC&quot;) + ylab(&quot;Fraction cell types in mIF&quot;) + 
  facet_wrap(.~matched_celltype, scales = &quot;free&quot;) </code></pre>
<p><img src="figure/01_celltype_correlations.Rmd/celltype-fractions-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-celltype-fractions-2">
Past versions of celltype-fractions-2.png
</button>
</p>
<div id="fig-celltype-fractions-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/4554a259201ef6a52df476c8208dede506b41b04/comparisons/docs/figure/01_celltype_correlations.Rmd/celltype-fractions-2.png" target="_blank">4554a25</a>
</td>
<td>
nilseling
</td>
<td>
2022-06-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/3a84dd86509fca85017d510c405b09733b2e3e26/comparisons/docs/figure/01_celltype_correlations.Rmd/celltype-fractions-2.png" target="_blank">3a84dd8</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/75d0a6df2a424569fc993d7f2034379ea37c1598/comparisons/docs/figure/01_celltype_correlations.Rmd/celltype-fractions-2.png" target="_blank">75d0a6d</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/2a2141368265bcc975aeb5543b40fad8d4b93375/comparisons/docs/figure/01_celltype_correlations.Rmd/celltype-fractions-2.png" target="_blank">2a21413</a>
</td>
<td>
nilseling
</td>
<td>
2022-04-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We will also now compute the correlation coefficients.</p>
<pre class="r"><code>combined_df %&gt;%
  group_by(matched_celltype) %&gt;%
  mutate(cor_all = cor(frac.x, frac.y)) %&gt;%
  ungroup() %&gt;%
  group_by(matched_celltype, patient_id) %&gt;%
  summarize(cor = cor(frac.x, frac.y),
            cor_all = mean(cor_all)) %&gt;%
  ggplot() +
    geom_boxplot(aes(matched_celltype, cor)) +
    geom_point(aes(matched_celltype, cor, color = patient_id)) +
    geom_point(aes(matched_celltype, cor_all), color = &quot;dark red&quot;, size = 3) +
    theme_classic(base_size = 20) + theme(axis.text.x = element_text(angle = 90, hjust = 1))</code></pre>
<pre><code>`summarise()` has grouped output by &#39;matched_celltype&#39;. You can override using
the `.groups` argument.</code></pre>
<pre><code>Warning: Removed 4 rows containing non-finite values (`stat_boxplot()`).</code></pre>
<pre><code>Warning: Removed 4 rows containing missing values (`geom_point()`).</code></pre>
<p><img src="figure/01_celltype_correlations.Rmd/correlation-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-correlation-1">
Past versions of correlation-1.png
</button>
</p>
<div id="fig-correlation-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/4554a259201ef6a52df476c8208dede506b41b04/comparisons/docs/figure/01_celltype_correlations.Rmd/correlation-1.png" target="_blank">4554a25</a>
</td>
<td>
nilseling
</td>
<td>
2022-06-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/3a84dd86509fca85017d510c405b09733b2e3e26/comparisons/docs/figure/01_celltype_correlations.Rmd/correlation-1.png" target="_blank">3a84dd8</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/75d0a6df2a424569fc993d7f2034379ea37c1598/comparisons/docs/figure/01_celltype_correlations.Rmd/correlation-1.png" target="_blank">75d0a6d</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/add09ed9175efe0986f12d7748eac4ca4d3fca7f/comparisons/docs/figure/01_celltype_correlations.Rmd/correlation-1.png" target="_blank">add09ed</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="absolute-number-of-cells" class="section level2">
<h2>Absolute number of cells</h2>
<p>We will now compare the absolute number of detected cell types.</p>
<pre class="r"><code>abs_IMC &lt;- cur_IMC %&gt;% as_tibble() %&gt;%
  group_by(sample_id, matched_celltype, .drop = FALSE) %&gt;%
  summarize(count = n()) </code></pre>
<pre><code>`summarise()` has grouped output by &#39;sample_id&#39;. You can override using the
`.groups` argument.</code></pre>
<pre class="r"><code>abs_mIF &lt;- cur_mIF %&gt;% as_tibble() %&gt;%
  group_by(sample_id, matched_celltype, .drop = FALSE) %&gt;%
  summarize(count = n()) </code></pre>
<pre><code>`summarise()` has grouped output by &#39;sample_id&#39;. You can override using the
`.groups` argument.</code></pre>
<pre class="r"><code>combined_df &lt;- merge(abs_IMC, abs_mIF, by=c(&quot;sample_id&quot;, &quot;matched_celltype&quot;), sort=FALSE)
combined_df$patient_id &lt;- sub(&quot;-&quot;, &quot;&quot;, str_extract(combined_df$sample_id, &quot;[0-9]{8}-&quot;))

ggplot(combined_df) + 
  geom_point(aes(count.x, count.y, color = patient_id)) + 
  geom_abline(slope = 1, intercept = 0, color = &quot;dark red&quot;) +
  geom_smooth(aes(count.x, count.y), method=&#39;lm&#39;, formula= y~x, fullrange=TRUE) + 
  theme_classic(base_size = 20) + xlab(&quot;Absolute cell types in IMC&quot;) + ylab(&quot;Absolute cell types in mIF&quot;) +
  facet_wrap(.~matched_celltype, scales = &quot;fixed&quot;) </code></pre>
<p><img src="figure/01_celltype_correlations.Rmd/unnamed-chunk-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/09d32f61a98b1b588620a3bd76317681b4a3743b/comparisons/docs/figure/01_celltype_correlations.Rmd/unnamed-chunk-1-1.png" target="_blank">09d32f6</a>
</td>
<td>
nilseling
</td>
<td>
2022-06-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(combined_df) + 
  geom_smooth(aes(count.x, count.y, color = patient_id),  se = FALSE, method=&#39;lm&#39;, formula= y~x, fullrange=TRUE) + 
  geom_point(aes(count.x, count.y, color = patient_id)) + 
  geom_abline(slope = 1, intercept = 0, color = &quot;dark red&quot;) +
  theme_classic(base_size = 20) + xlab(&quot;Absolute cell types in IMC&quot;) + ylab(&quot;Absolute cell types in mIF&quot;) + 
  facet_wrap(.~matched_celltype, scales = &quot;free&quot;) </code></pre>
<p><img src="figure/01_celltype_correlations.Rmd/unnamed-chunk-1-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-2">
Past versions of unnamed-chunk-1-2.png
</button>
</p>
<div id="fig-unnamed-chunk-1-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/09d32f61a98b1b588620a3bd76317681b4a3743b/comparisons/docs/figure/01_celltype_correlations.Rmd/unnamed-chunk-1-2.png" target="_blank">09d32f6</a>
</td>
<td>
nilseling
</td>
<td>
2022-06-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="perform-differential-abundance-analysis"
class="section level2">
<h2>Perform differential abundance analysis</h2>
<p>We will next test for systematic biases in cell type abundances
between the two technologies.</p>
<pre class="r"><code>library(edgeR)
library(ggrepel)

abundances_IMC &lt;- cur_IMC %&gt;% as_tibble() %&gt;%
  group_by(sample_id, matched_celltype) %&gt;%
  summarize(count = n()) %&gt;%
  pivot_wider(names_from = sample_id, 
              values_from = count, values_fill = 0) %&gt;%
  as.data.frame()
rownames(abundances_IMC) &lt;- abundances_IMC$matched_celltype
abundances_IMC$matched_celltype &lt;- NULL
meta_IMC &lt;- data.frame(sample_id = colnames(abundances_IMC),
                       patient_id = sub(&quot;-&quot;, &quot;&quot;, str_extract(colnames(abundances_IMC), &quot;[0-9]{8}-&quot;)),
                       modality = &quot;IMC&quot;)

abundances_IF &lt;- cur_mIF %&gt;% as_tibble() %&gt;%
  group_by(sample_id, matched_celltype) %&gt;%
  summarize(count = n()) %&gt;%
  pivot_wider(names_from = sample_id, 
              values_from = count, values_fill = 0) %&gt;%
  as.data.frame()
rownames(abundances_IF) &lt;- abundances_IF$matched_celltype
abundances_IF$matched_celltype &lt;- NULL
meta_IF &lt;- data.frame(sample_id = colnames(abundances_IF),
                       patient_id = sub(&quot;-&quot;, &quot;&quot;, str_extract(colnames(abundances_IMC), &quot;[0-9]{8}-&quot;)),
                       modality = &quot;IF&quot;)

colnames(abundances_IMC) &lt;- paste(abundances_IMC, &quot;IMC&quot;)
colnames(abundances_IF) &lt;- paste(abundances_IF, &quot;IF&quot;)

combined_df &lt;- cbind(abundances_IMC, abundances_IF[rownames(abundances_IMC),])
combined_meta &lt;- rbind(meta_IMC, meta_IF)

# DA testing
y.ab &lt;- DGEList(combined_df, samples=combined_meta, group = combined_meta$modality)

# Define the design matrix for the model: Add Patient ID to the model to account for matched design, add Location to account for different tumor regions, add tissue type as an additive factor
design &lt;- model.matrix(~sample_id + modality, data = y.ab$samples)

# Estimate NB and QL dispersion
y.ab &lt;- estimateDisp(y.ab, design, trend=&quot;none&quot;)
summary(y.ab$common.dispersion)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.2581  0.2581  0.2581  0.2581  0.2581  0.2581 </code></pre>
<pre class="r"><code>plotBCV(y.ab, cex=1)</code></pre>
<p><img src="figure/01_celltype_correlations.Rmd/test-DA-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-test-DA-1">
Past versions of test-DA-1.png
</button>
</p>
<div id="fig-test-DA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/4554a259201ef6a52df476c8208dede506b41b04/comparisons/docs/figure/01_celltype_correlations.Rmd/test-DA-1.png" target="_blank">4554a25</a>
</td>
<td>
nilseling
</td>
<td>
2022-06-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/75d0a6df2a424569fc993d7f2034379ea37c1598/comparisons/docs/figure/01_celltype_correlations.Rmd/test-DA-1.png" target="_blank">75d0a6d</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>fit.ab &lt;- glmQLFit(y.ab, design, robust=TRUE, abundance.trend=FALSE)
summary(fit.ab$var.prior)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.5619  0.5619  0.5619  0.5619  0.5619  0.5619 </code></pre>
<pre class="r"><code>plotQLDisp(fit.ab, cex=1)</code></pre>
<p><img src="figure/01_celltype_correlations.Rmd/test-DA-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-test-DA-2">
Past versions of test-DA-2.png
</button>
</p>
<div id="fig-test-DA-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/4554a259201ef6a52df476c8208dede506b41b04/comparisons/docs/figure/01_celltype_correlations.Rmd/test-DA-2.png" target="_blank">4554a25</a>
</td>
<td>
nilseling
</td>
<td>
2022-06-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/75d0a6df2a424569fc993d7f2034379ea37c1598/comparisons/docs/figure/01_celltype_correlations.Rmd/test-DA-2.png" target="_blank">75d0a6d</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Test for differential abundance of cell types in PT vs met tissue and plot
res &lt;- glmQLFTest(fit.ab, coef=colnames(fit.ab)[ncol(fit.ab)])
summary(decideTests(res))</code></pre>
<pre><code>       modalityIMC
Down             1
NotSig           5
Up               2</code></pre>
<pre class="r"><code>DA &lt;- topTags(res)$table

DA$celltype &lt;- rownames(DA)
DA$sign &lt;- DA$FDR&lt;0.05
DA</code></pre>
<pre><code>                 logFC   logCPM          F       PValue          FDR   celltype
DC          2.11526359 14.54511 56.7802760 4.246216e-09 3.396973e-08         DC
Bcell      -0.90478794 15.35011  9.2317191 4.197643e-03 1.679057e-02      Bcell
BnTcell     0.75786303 15.69220  8.3879988 6.417700e-03 1.711387e-02    BnTcell
MacCD163   -0.35934699 16.46330  4.4808165 4.060873e-02 8.121746e-02   MacCD163
Neutrophil  0.56682930 14.66975  3.2710313 7.873150e-02 1.259704e-01 Neutrophil
other       0.10792410 17.69369  1.1617524 2.876222e-01 3.834962e-01      other
Tumor      -0.09034737 18.50573  0.5744237 4.529916e-01 5.177047e-01      Tumor
Tcell      -0.02709685 17.50460  0.1734418 6.793188e-01 6.793188e-01      Tcell
            sign
DC          TRUE
Bcell       TRUE
BnTcell     TRUE
MacCD163   FALSE
Neutrophil FALSE
other      FALSE
Tumor      FALSE
Tcell      FALSE</code></pre>
<pre class="r"><code># Volcano plot
ggplot(DA) + geom_point(aes(logFC, -log10(FDR), color = sign)) +
  geom_label_repel(aes(logFC, -log10(FDR), label = celltype)) + 
  theme_classic()</code></pre>
<p><img src="figure/01_celltype_correlations.Rmd/test-DA-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-test-DA-3">
Past versions of test-DA-3.png
</button>
</p>
<div id="fig-test-DA-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/4554a259201ef6a52df476c8208dede506b41b04/comparisons/docs/figure/01_celltype_correlations.Rmd/test-DA-3.png" target="_blank">4554a25</a>
</td>
<td>
nilseling
</td>
<td>
2022-06-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/75d0a6df2a424569fc993d7f2034379ea37c1598/comparisons/docs/figure/01_celltype_correlations.Rmd/test-DA-3.png" target="_blank">75d0a6d</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># MA plot
ggplot(DA) + geom_point(aes(logCPM, logFC, color = sign)) +
  geom_label_repel(aes(logCPM, logFC, label = celltype)) + 
  geom_abline(intercept = 0, slope = 0, color = &quot;dark red&quot;) +
  theme_classic(base_size = 20)</code></pre>
<p><img src="figure/01_celltype_correlations.Rmd/test-DA-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-test-DA-4">
Past versions of test-DA-4.png
</button>
</p>
<div id="fig-test-DA-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/4554a259201ef6a52df476c8208dede506b41b04/comparisons/docs/figure/01_celltype_correlations.Rmd/test-DA-4.png" target="_blank">4554a25</a>
</td>
<td>
nilseling
</td>
<td>
2022-06-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/3a84dd86509fca85017d510c405b09733b2e3e26/comparisons/docs/figure/01_celltype_correlations.Rmd/test-DA-4.png" target="_blank">3a84dd8</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/75d0a6df2a424569fc993d7f2034379ea37c1598/comparisons/docs/figure/01_celltype_correlations.Rmd/test-DA-4.png" target="_blank">75d0a6d</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="compare-expression-of-cell-types" class="section level2">
<h2>Compare expression of cell types</h2>
<p>As heatmaps</p>
</div>
<div id="marker-expression" class="section level2">
<h2>Marker expression</h2>
<p>Here, we will min-max scale the marker expression per image and
compare binned rank averages.</p>
<pre class="r"><code>cur_IMC &lt;-  lapply(unique(sce_IMC$sample_id), function(k){
  cur_sce_IMC &lt;- sce_IMC[,sce_IMC$sample_id == k]
  cur_sce_IMC &lt;- cur_sce_IMC[c(&quot;CD15&quot;, &quot;Ecad&quot;, &quot;CD3&quot;, &quot;CD11c&quot;, &quot;CD20&quot;, &quot;CD163&quot;),]
  rownames(cur_sce_IMC) &lt;- c(&quot;CD15&quot;, &quot;CK&quot;, &quot;CD3&quot;, &quot;CD11c&quot;, &quot;CD20&quot;, &quot;CD163&quot;)
  assay(cur_sce_IMC, &quot;scaled&quot;) &lt;- (assay(cur_sce_IMC, &quot;exprs&quot;) - 
                                     rowMin(assay(cur_sce_IMC, &quot;exprs&quot;))) / 
                                  (rowMax(assay(cur_sce_IMC, &quot;exprs&quot;)) - 
                                     rowMin(assay(cur_sce_IMC, &quot;exprs&quot;)))
  
  cur_aggr_IMC &lt;- lapply(rownames(sce_mIF), function(x){
    cur_marker_IMC &lt;- assay(cur_sce_IMC, &quot;scaled&quot;)[x,]
    cur_out &lt;- aggregate(cur_marker_IMC, by = list(ntile(cur_marker_IMC, 100)), mean)
    cur_out$marker &lt;- x
    return(cur_out)
  })
  cur_aggr_IMC &lt;- do.call(rbind, cur_aggr_IMC)
  cur_aggr_IMC$sample_id &lt;- k 
  
  return(cur_aggr_IMC)
})
cur_IMC &lt;- do.call(rbind, cur_IMC)

cur_IF &lt;- lapply(unique(sce_mIF$sample_id), function(k){
  cur_sce_IF &lt;- sce_mIF[,sce_mIF$sample_id == k]
  assay(cur_sce_IF, &quot;scaled&quot;) &lt;- (assay(cur_sce_IF, &quot;exprs&quot;) - 
                                     rowMin(assay(cur_sce_IF, &quot;exprs&quot;))) / 
                                  (rowMax(assay(cur_sce_IF, &quot;exprs&quot;)) - 
                                     rowMin(assay(cur_sce_IF, &quot;exprs&quot;)))
  
  cur_aggr_IF &lt;- lapply(rownames(sce_mIF), function(x){
    cur_marker_IF &lt;- assay(cur_sce_IF, &quot;scaled&quot;)[x,]
    cur_out &lt;- aggregate(cur_marker_IF, by = list(ntile(cur_marker_IF, 100)), mean)
    cur_out$marker &lt;- x
    return(cur_out)
  })
  cur_aggr_IF &lt;- do.call(rbind, cur_aggr_IF)
  cur_aggr_IF$sample_id &lt;- k 
  
  return(cur_aggr_IF)
})

cur_IF &lt;- do.call(rbind, cur_IF)

combined_df &lt;- merge(cur_IMC, cur_IF, by = c(&quot;Group.1&quot;, &quot;marker&quot;, &quot;sample_id&quot;), sort = FALSE)</code></pre>
<p>Visualize the results</p>
<pre class="r"><code>ggplot(combined_df) +
  geom_line(aes(x.x, x.y, color = sample_id)) + 
  facet_wrap(. ~ marker) + 
  geom_abline(intercept = 0, slope = 1, color = &quot;dark red&quot;) + 
  theme(legend.position = &quot;none&quot;) + xlab(&quot;Expression IMC&quot;) + ylab(&quot;Expression IF&quot;)</code></pre>
<p><img src="figure/01_celltype_correlations.Rmd/marker-expression-viz-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-marker-expression-viz-1">
Past versions of marker-expression-viz-1.png
</button>
</p>
<div id="fig-marker-expression-viz-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/4554a259201ef6a52df476c8208dede506b41b04/comparisons/docs/figure/01_celltype_correlations.Rmd/marker-expression-viz-1.png" target="_blank">4554a25</a>
</td>
<td>
nilseling
</td>
<td>
2022-06-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/75d0a6df2a424569fc993d7f2034379ea37c1598/comparisons/docs/figure/01_celltype_correlations.Rmd/marker-expression-viz-1.png" target="_blank">75d0a6d</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/add09ed9175efe0986f12d7748eac4ca4d3fca7f/comparisons/docs/figure/01_celltype_correlations.Rmd/marker-expression-viz-1.png" target="_blank">add09ed</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="visualization-on-images" class="section level2">
<h2>Visualization on images</h2>
<p>We will next plot the location of celltypes on the images.</p>
<pre class="r"><code>library(cytomapper)
images_mIF &lt;- readRDS(&quot;/Volumes/immucan_volume/processed_data/Panel_1/2022_WORKFLOW/mIF/Rout/images.rds&quot;)
images_IMC &lt;- readRDS(&quot;/Volumes/immucan_volume/processed_data/Panel_1/2022_WORKFLOW/IMC/Rout/images.rds&quot;)

for (i in unique(sce_IMC$sample_id)) {
  
  if (! i %in% names(images_mIF)) next
  
  cur_IMC &lt;- plotPixels(images_IMC[i], 
                        colour_by = c(&quot;Ecad&quot;, &quot;CD3&quot;, &quot;CD20&quot;, &quot;CD163&quot;, &quot;CD15&quot;, &quot;CD11c&quot;),
                        return_images = TRUE, image_title = NULL, 
                        scale_bar = NULL, legend = NULL,
                        bcg = list(Ecad = c(0, 5, 1),
                                   CD3 = c(0, 5, 1),
                                   CD20 = c(0, 5, 1),
                                   CD163 = c(0, 5, 1),
                                   CD15 = c(0, 5, 1),
                                   CD11c = c(0, 5, 1)))
  
    cur_mIF &lt;- plotPixels(images_mIF[i], 
                        colour_by = c(&quot;CK&quot;, &quot;CD3&quot;, &quot;CD20&quot;, &quot;CD163&quot;, &quot;CD15&quot;, &quot;CD11c&quot;),
                        return_images = TRUE, image_title = NULL, 
                        scale_bar = NULL, legend = NULL,
                        bcg = list(Ecad = c(0, 5, 1),
                                   CD3 = c(0, 5, 1),
                                   CD20 = c(0, 5, 1),
                                   CD163 = c(0, 5, 1),
                                   CD15 = c(0, 5, 1),
                                   CD11c = c(0, 5, 1)))
    
    # Add points to IMC
    cur_sce_IMC &lt;- sce_IMC[,sce_IMC$sample_id == i]
    cur_sce_mIF &lt;- sce_mIF[,sce_mIF$sample_id == i]
    pdf(file = paste0(&quot;~/SIB Swiss Institute of Bioinformatics/Robin Liechti - Workflow2021/comparisons/images_celltypes/&quot;, i, &quot;.pdf&quot;))
    plot(cur_IMC$images[[1]])
    points(cur_sce_IMC$Pos_X, cur_sce_IMC$Pos_Y, 
           col = &quot;white&quot;, 
           pch = 16, cex = 1.1)
    points(cur_sce_IMC$Pos_X, cur_sce_IMC$Pos_Y, 
           col = metadata(cur_sce_IMC)$colour_vectors$matched_celltype[cur_sce_IMC$matched_celltype], 
           pch = 16, cex = 0.9)
    plot(cur_mIF$images[[1]])
    points(cur_sce_mIF$nucleus.x, cur_sce_mIF$nucleus.y, 
           col = &quot;white&quot;, 
           pch = 16, cex = 1.1)
    points(cur_sce_mIF$nucleus.x, cur_sce_mIF$nucleus.y, 
           col = metadata(cur_sce_mIF)$color_vectors$matched_celltype[cur_sce_mIF$matched_celltype], 
           pch = 16, cex = 0.9)
    dev.off()
  
}

# Write out legend
pdf(file = paste0(&quot;~/SIB Swiss Institute of Bioinformatics/Robin Liechti - Workflow2021/comparisons/images_celltypes/legend.pdf&quot;))
plot(1:8, y = rep(1, 8), col = metadata(sce_IMC)$color_vectors$matched_celltype, pch = 16, cex = 2)
text(1:8, y = rep(0.8, 8), labels = names(metadata(sce_IMC)$color_vectors$matched_celltype))
text(2, y = 1.2, labels = &quot;Celltypes&quot;)
plot(1:6, y = rep(1, 6), col = c(&quot;red&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;cyan&quot;, &quot;magenta&quot;, &quot;yellow&quot;), pch = 16, cex = 2)
text(1:6, y = rep(0.8, 6), labels = c(&quot;Ecad&quot;, &quot;CD3&quot;, &quot;CD20&quot;, &quot;CD163&quot;, &quot;CD15&quot;, &quot;CD11c&quot;))
text(2, y = 1.2, labels = &quot;Marker expression&quot;)
dev.off()</code></pre>
</div>
<div id="save-out-data" class="section level2">
<h2>Save out data</h2>
<p>For cell type comparison I will now write out the x/y cooridnates and
cell labels for the IMC and mIF data.</p>
<pre class="r"><code>for (i in unique(sce_IMC$sample_id)) {
    cur_sce_IMC &lt;- sce_IMC[,sce_IMC$sample_id == i]
    cur_sce_mIF &lt;- sce_mIF[,sce_mIF$sample_id == i]
   
    write_tsv(as.data.frame(colData(cur_sce_IMC)), paste0(&quot;~/SIB Swiss Institute of Bioinformatics/Robin Liechti - Workflow2021/Zurich_data/data_for_comparisons/IMC_cells/&quot;, i, &quot;.tsv&quot;))   
        write_tsv(as.data.frame(colData(cur_sce_mIF)), paste0(&quot;~/SIB Swiss Institute of Bioinformatics/Robin Liechti - Workflow2021/Zurich_data/data_for_comparisons/mIF_cells//&quot;, i, &quot;.tsv&quot;))   
}</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ggrepel_0.9.2               edgeR_3.40.0               
 [3] limma_3.54.0                forcats_0.5.2              
 [5] stringr_1.4.1               dplyr_1.0.10               
 [7] purrr_0.3.5                 readr_2.1.3                
 [9] tidyr_1.2.1                 tibble_3.1.8               
[11] ggplot2_3.4.0               tidyverse_1.3.2            
[13] SingleCellExperiment_1.20.0 SummarizedExperiment_1.28.0
[15] Biobase_2.58.0              GenomicRanges_1.50.0       
[17] GenomeInfoDb_1.34.1         IRanges_2.32.0             
[19] S4Vectors_0.36.0            BiocGenerics_0.44.0        
[21] MatrixGenerics_1.10.0       matrixStats_0.62.0         

loaded via a namespace (and not attached):
 [1] nlme_3.1-160           bitops_1.0-7           fs_1.5.2              
 [4] bit64_4.0.5            lubridate_1.9.0        httr_1.4.4            
 [7] rprojroot_2.0.3        tools_4.2.1            backports_1.4.1       
[10] bslib_0.4.1            utf8_1.2.2             R6_2.5.1              
[13] mgcv_1.8-41            DBI_1.1.3              colorspace_2.0-3      
[16] withr_2.5.0            tidyselect_1.2.0       bit_4.0.4             
[19] compiler_4.2.1         git2r_0.30.1           cli_3.4.1             
[22] rvest_1.0.3            xml2_1.3.3             DelayedArray_0.24.0   
[25] labeling_0.4.2         sass_0.4.2             scales_1.2.1          
[28] digest_0.6.30          rmarkdown_2.17         XVector_0.38.0        
[31] pkgconfig_2.0.3        htmltools_0.5.3        highr_0.9             
[34] dbplyr_2.2.1           fastmap_1.1.0          rlang_1.0.6           
[37] readxl_1.4.1           rstudioapi_0.14        farver_2.1.1          
[40] jquerylib_0.1.4        generics_0.1.3         jsonlite_1.8.3        
[43] vroom_1.6.0            googlesheets4_1.0.1    RCurl_1.98-1.9        
[46] magrittr_2.0.3         GenomeInfoDbData_1.2.9 Matrix_1.5-1          
[49] Rcpp_1.0.9             munsell_0.5.0          fansi_1.0.3           
[52] lifecycle_1.0.3        stringi_1.7.8          whisker_0.4           
[55] yaml_2.3.6             zlibbioc_1.44.0        grid_4.2.1            
[58] parallel_4.2.1         promises_1.2.0.1       crayon_1.5.2          
[61] lattice_0.20-45        splines_4.2.1          haven_2.5.1           
[64] hms_1.1.2              locfit_1.5-9.6         knitr_1.40            
[67] pillar_1.8.1           reprex_2.0.2           glue_1.6.2            
[70] evaluate_0.18          modelr_0.1.9           vctrs_0.5.0           
[73] tzdb_0.3.0             httpuv_1.6.6           cellranger_1.1.0      
[76] gtable_0.3.1           assertthat_0.2.1       cachem_1.0.6          
[79] xfun_0.34              broom_1.0.1            later_1.3.0           
[82] googledrive_2.0.0      gargle_1.2.1           workflowr_1.7.0       
[85] timechange_0.1.1       ellipsis_0.3.2        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
