<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Label cell types</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.23/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">IMMUcanWorkflow2022</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Label cell types</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-06-23
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>Analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220321code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220321)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220321code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220321)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomImmucanWP7IMMUcanWorkflow2021tree2a51289e9899968d8af665f68150b3cbf7eb3653targetblank2a51289a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/tree/2a51289e9899968d8af665f68150b3cbf7eb3653" target="_blank">2a51289</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomImmucanWP7IMMUcanWorkflow2021tree2a51289e9899968d8af665f68150b3cbf7eb3653targetblank2a51289a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/tree/2a51289e9899968d8af665f68150b3cbf7eb3653" target="_blank">2a51289</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    Alignment/
    Ignored:    IMC/.DS_Store
    Ignored:    IMC/Analysis/.DS_Store
    Ignored:    IMC/Analysis/.Rproj.user/
    Ignored:    comparisons/.Rproj.user/
    Ignored:    comparisons/analysis/figure/
    Ignored:    mIF/.DS_Store
    Ignored:    mIF/Analysis/.Rproj.user/

Untracked files:
    Untracked:  IMC/Analysis/code/transfer_labels.R
    Untracked:  mIF/Analysis/docs/zzz_read_data.html

Unstaged changes:
    Modified:   IMC/Analysis/analysis/01_read_data.Rmd
    Modified:   IMC/Analysis/analysis/05_labelling.Rmd
    Modified:   comparisons/analysis/01_celltype_correlations.Rmd
    Modified:   mIF/Analysis/analysis/zzz_read_data.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>IMC/Analysis/analysis/05_labelling.Rmd</code>) and HTML
(<code>IMC/Analysis/docs/05_labelling.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/8d1f8ec230505f81891267366712aeb5847bda1a/IMC/Analysis/analysis/05_labelling.Rmd" target="_blank">8d1f8ec</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-30
</td>
<td>
Recompiled workflow
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ImmucanWP7/IMMUcanWorkflow2021/8d1f8ec230505f81891267366712aeb5847bda1a/IMC/Analysis/docs/05_labelling.html" target="_blank">8d1f8ec</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-30
</td>
<td>
Recompiled workflow
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ImmucanWP7/IMMUcanWorkflow2021/f7cc35b2360b342abaa00a75373b5f8a03f47633/IMC/Analysis/docs/05_labelling.html" target="_blank">f7cc35b</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-11
</td>
<td>
Recompiled scripts
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/c22d4e1fe74fdd577574bf3c0c514bb5df0a06fa/IMC/Analysis/analysis/05_labelling.Rmd" target="_blank">c22d4e1</a>
</td>
<td>
nilseling
</td>
<td>
2022-04-29
</td>
<td>
Added more projects
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Here, we will manually label cells based on their cell type. This is
done based on the labelling scheme in …</p>
<div id="load-data" class="section level2">
<h2>Load data</h2>
<p>First, we will load the previously generated
<code>SingleCellExperiment</code> and <code>CytoImageList</code>
objects.</p>
<pre class="r"><code>library(cytomapper)
library(stringr)

sce &lt;- readRDS(&quot;/Volumes/immucan_volume/processed_data/Panel_1/2022_WORKFLOW/IMC/Rout/sce.rds&quot;)
images &lt;- readRDS(&quot;/Volumes/immucan_volume/processed_data/Panel_1/2022_WORKFLOW/IMC/Rout/images.rds&quot;)
masks &lt;- readRDS(&quot;/Volumes/immucan_volume/processed_data/Panel_1/2022_WORKFLOW/IMC/Rout/masks.rds&quot;)</code></pre>
</div>
<div id="label-images" class="section level2">
<h2>Label images</h2>
<p>We will now label cells on images and save them.</p>
<pre class="r"><code>images &lt;- CytoImageList(images, on_disk = FALSE)
masks &lt;- CytoImageList(masks, on_disk = FALSE)

# Change display name
sce$ID &lt;- paste0(sce$indication, &quot;_&quot;, sce$patient_id, &quot;_&quot;, sce$ROI)
cur_df &lt;- colData(sce)[,c(&quot;sample_id&quot;, &quot;ID&quot;)]
cur_mcols &lt;- mcols(images)

cur_mcols &lt;- merge(cur_mcols, cur_df, by = &quot;sample_id&quot;, sort = FALSE)
cur_mcols &lt;- unique(cur_mcols)

all.equal(cur_mcols$sample_id, mcols(images)$sample_id)</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code>mcols(images) &lt;- mcols(masks) &lt;- cur_mcols 

if (interactive()) {
  cytomapperShiny(object = sce, mask = masks, image = images,
                  cell_id = &quot;ObjectNumber&quot;, img_id = &quot;ID&quot;)
}</code></pre>
</div>
<div id="quality-control" class="section level2">
<h2>Quality control</h2>
<p>We next check for labelling quality.</p>
<pre class="r"><code>library(SingleCellExperiment)
label.files &lt;- list.files(&quot;/Volumes/immucan_volume/processed_data/Panel_1/2022_WORKFLOW/IMC/Rout/gated_cells/&quot;, 
                          full.names = TRUE, pattern = &quot;.rds$&quot;)

# Read in SCE objects
cur_sces &lt;- lapply(label.files, readRDS)
cur_sces &lt;- cur_sces[-108] 

# Merge SCE objects
cur_sce &lt;- do.call(&quot;cbind&quot;, cur_sces)

# now we check if the celltypes were labelled uniquely
names &lt;- unique(cur_sce$cytomapper_CellLabel)
unique(names)</code></pre>
<pre><code> [1] &quot;B_cell&quot;     &quot;BnCD4&quot;      &quot;BnCD8&quot;      &quot;CD38&quot;       &quot;CD4&quot;       
 [6] &quot;CD8&quot;        &quot;DC&quot;         &quot;HLADR&quot;      &quot;Macrophage&quot; &quot;Neutrophil&quot;
[11] &quot;NK_cell&quot;    &quot;pDC&quot;        &quot;Stroma&quot;     &quot;Treg&quot;       &quot;Tumor&quot;     </code></pre>
<pre class="r"><code>table(cur_sce$cytomapper_CellLabel, cur_sce$indication)</code></pre>
<pre><code>            
              BCC  CRC NSCLC  RCC SCCHN
  B_cell      310   66   586  990   915
  BnCD4       374  120   513  540   772
  BnCD8       191  268   263  246   663
  CD38         55  262   701  417   506
  CD4          66  402   168  234   483
  CD8          86  142  1301  476   143
  DC           41  116   386  191   234
  HLADR       116  183   641  795   363
  Macrophage  147  362   483  373   346
  Neutrophil   49   91   673  388   349
  NK_cell      52   20    33  113   120
  pDC           2   30   271   48    52
  Stroma      180  135   176  530   463
  Treg         30   11    53    5   125
  Tumor      1193 1149  2761 3810  2201</code></pre>
<p>We next need to remove cells that were assigned to different cell
types. For these cell, the classification uncertainty will be
highest.</p>
<pre class="r"><code>cur_vec &lt;- cur_sce$cytomapper_CellLabel

cur_tab &lt;- unclass(table(colnames(cur_sce), cur_sce$cytomapper_CellLabel))
cur_labels &lt;- rep(&quot;doublets&quot;, nrow(cur_tab))

# Single assignments
cur_labels[rowSums(cur_tab) == 1] &lt;- colnames(cur_tab)[apply(cur_tab[rowSums(cur_tab) == 1,], 1,  which.max)]

table(cur_labels)</code></pre>
<pre><code>cur_labels
    B_cell      BnCD4      BnCD8       CD38        CD4        CD8         DC 
      2779       2293       1616       1554       1167       1873        810 
  doublets      HLADR Macrophage Neutrophil    NK_cell        pDC     Stroma 
      1819       1443       1430       1276        246        312       1440 
      Treg      Tumor 
       155      10061 </code></pre>
<pre class="r"><code># Remove doublets
names(cur_labels) &lt;- rownames(cur_tab)
cur_labels &lt;- cur_labels[cur_labels != &quot;doublets&quot;]
table(cur_labels)</code></pre>
<pre><code>cur_labels
    B_cell      BnCD4      BnCD8       CD38        CD4        CD8         DC 
      2779       2293       1616       1554       1167       1873        810 
     HLADR Macrophage Neutrophil    NK_cell        pDC     Stroma       Treg 
      1443       1430       1276        246        312       1440        155 
     Tumor 
     10061 </code></pre>
<pre class="r"><code># Store in SingleCellExperiment
sce_labels &lt;- rep(&quot;unlabelled&quot;, ncol(sce))
names(sce_labels) &lt;- colnames(sce)
sce_labels[names(cur_labels)] &lt;- cur_labels
sce$cell_labels &lt;- sce_labels

table(sce$cell_labels, sce$indication)</code></pre>
<pre><code>            
               BCC   CRC NSCLC   RCC SCCHN
  B_cell       294    66   583   926   910
  BnCD4        363   120   512   528   770
  BnCD8        187   268   261   239   661
  CD38          53   262   619   308   312
  CD4           66   374   151   203   373
  CD8           77   125  1169   383   119
  DC            40   110   351   110   199
  HLADR        115   153   542   363   270
  Macrophage   145   326   437   261   261
  Neutrophil    20    91   566   383   216
  NK_cell       50     3    18    67   108
  pDC            2     2   269    28    11
  Stroma       180   100   174   525   461
  Treg          29    11    52     5    58
  Tumor       1183  1109  2418  3150  2201
  unlabelled 12723 19720 19487 12865 13909</code></pre>
</div>
<div id="define-colour-vector" class="section level2">
<h2>Define colour vector</h2>
<p>In this step, we will define a colour vector for all further
visualization purposes.</p>
<pre class="r"><code>cell_types &lt;- c(&quot;B_cell&quot;, &quot;CD38&quot;, &quot;CD4&quot;, &quot;CD8&quot;, &quot;Treg&quot;, &quot;BnCD4&quot;, &quot;BnCD8&quot;,
                &quot;DC&quot;, &quot;HLADR&quot;, &quot;Macrophage&quot;, &quot;Neutrophil&quot;, &quot;NK_cell&quot;, &quot;pDC&quot;, &quot;Stroma&quot;, 
                &quot;Tumor&quot;, &quot;unlabelled&quot;)

names(cell_types) &lt;- cell_types
cell_types[&quot;B_cell&quot;] &lt;- &quot;red3&quot;
cell_types[&quot;CD38&quot;] &lt;- &quot;goldenrod2&quot;
cell_types[&quot;CD4&quot;] &lt;- &quot;palevioletred1&quot;
cell_types[&quot;CD8&quot;] &lt;- &quot;darkorchid&quot;
cell_types[&quot;Treg&quot;] &lt;- &quot;maroon4&quot;
cell_types[&quot;BnCD4&quot;] &lt;- &quot;darkorange4&quot;
cell_types[&quot;BnCD8&quot;] &lt;- &quot;red4&quot;
cell_types[&quot;DC&quot;] &lt;- &quot;green4&quot;
cell_types[&quot;HLADR&quot;] &lt;- &quot;darkgreen&quot;
cell_types[&quot;Macrophage&quot;] &lt;- &quot;green&quot;
cell_types[&quot;Neutrophil&quot;] &lt;- &quot;blue1&quot;
cell_types[&quot;NK_cell&quot;] &lt;- &quot;yellow&quot;
cell_types[&quot;pDC&quot;] &lt;- &quot;deepskyblue&quot;
cell_types[&quot;Stroma&quot;] &lt;- &quot;tomato&quot;
cell_types[&quot;Tumor&quot;] &lt;- &quot;sienna4&quot;
cell_types[&quot;unlabelled&quot;] &lt;- &quot;gray&quot;

metadata(sce)$colour_vectors$cell_types &lt;- cell_types

pie(c(rep(1,length(metadata(sce)$colour_vectors$cell_types))),
    col = metadata(sce)$colour_vectors$cell_types,
    labels = names(metadata(sce)$colour_vectors$cell_types))</code></pre>
<p><img src="figure/05_labelling.Rmd/define-color-vector-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-define-color-vector-1">
Past versions of define-color-vector-1.png
</button>
</p>
<div id="fig-define-color-vector-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/f7cc35b2360b342abaa00a75373b5f8a03f47633/IMC/Analysis/docs/figure/05_labelling.Rmd/define-color-vector-1.png" target="_blank">f7cc35b</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="quality-control-of-labelling-results" class="section level2">
<h2>Quality control of labelling results</h2>
<p>Here, we will check a number of statistics to assess labelling
quality.</p>
<pre class="r"><code>library(dplyr)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:Biobase&#39;:

    combine</code></pre>
<pre><code>The following objects are masked from &#39;package:GenomicRanges&#39;:

    intersect, setdiff, union</code></pre>
<pre><code>The following object is masked from &#39;package:GenomeInfoDb&#39;:

    intersect</code></pre>
<pre><code>The following objects are masked from &#39;package:IRanges&#39;:

    collapse, desc, intersect, setdiff, slice, union</code></pre>
<pre><code>The following objects are masked from &#39;package:S4Vectors&#39;:

    first, intersect, rename, setdiff, setequal, union</code></pre>
<pre><code>The following objects are masked from &#39;package:BiocGenerics&#39;:

    combine, intersect, setdiff, union</code></pre>
<pre><code>The following object is masked from &#39;package:matrixStats&#39;:

    count</code></pre>
<pre><code>The following object is masked from &#39;package:EBImage&#39;:

    combine</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code># 1. How many cells of how many images are labelled

# Percent cells labelled
as_tibble(colData(sce)) %&gt;% 
  summarise(labelled_cells = sum(cell_labels != &quot;unlabelled&quot;)/n()) </code></pre>
<pre><code># A tibble: 1 × 1
  labelled_cells
           &lt;dbl&gt;
1          0.266</code></pre>
<pre class="r"><code># Percent images labelled
as_tibble(colData(sce)) %&gt;% 
  group_by(sample_id) %&gt;%
  summarise(labelled_cells = sum(cell_labels != &quot;unlabelled&quot;)) %&gt;%
  ungroup() %&gt;%
  summarise(labelled_images = sum(labelled_cells != 0)/n())</code></pre>
<pre><code># A tibble: 1 × 1
  labelled_images
            &lt;dbl&gt;
1           0.769</code></pre>
<pre class="r"><code># Percent patients labelled
as_tibble(colData(sce)) %&gt;% 
  group_by(patient_id) %&gt;%
  summarise(labelled_cells = sum(cell_labels != &quot;unlabelled&quot;)) %&gt;%
  ungroup() %&gt;%
  summarise(labelled_samples = sum(labelled_cells != 0)/n())</code></pre>
<pre><code># A tibble: 1 × 1
  labelled_samples
             &lt;dbl&gt;
1              0.8</code></pre>
<pre class="r"><code># Percent of cells labelled per image
as_tibble(colData(sce)) %&gt;% 
  group_by(sample_id) %&gt;%
  summarise(labelled_cells = sum(cell_labels != &quot;unlabelled&quot;)/n(),
            number_cells = n()) %&gt;%
  as.data.frame()</code></pre>
<pre><code>                                                 sample_id labelled_cells
1       IMMUcan_2022_WFLOW_10034294-GI-VAR-TIS-UNST-03_001    0.000000000
2       IMMUcan_2022_WFLOW_10034294-GI-VAR-TIS-UNST-03_002    0.000000000
3       IMMUcan_2022_WFLOW_10034294-GI-VAR-TIS-UNST-03_003    0.000000000
4       IMMUcan_2022_WFLOW_10034294-GI-VAR-TIS-UNST-03_004    0.000000000
5  IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_001    0.678517168
6  IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_002    0.429128739
7  IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_003    0.285918004
8  IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_004    0.120327519
9  IMMUcan_2022_WFLOW_10063152-SPECT-VAR-TIS-01-IMC-01_001    0.000000000
10 IMMUcan_2022_WFLOW_10063152-SPECT-VAR-TIS-01-IMC-01_002    0.000000000
11 IMMUcan_2022_WFLOW_10063152-SPECT-VAR-TIS-01-IMC-01_003    0.000000000
12 IMMUcan_2022_WFLOW_10063152-SPECT-VAR-TIS-01-IMC-01_004    0.000000000
13 IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_001    0.635299625
14 IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_002    0.096670247
15 IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_003    0.053285049
16 IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_004    0.419157364
17 IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_005    0.755338078
18 IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_006    0.452099738
19 IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_007    0.090077978
20 IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_008    0.274247492
21 IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_005    0.217043941
22 IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_006    0.134414832
23 IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_007    0.361654804
24 IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_008    0.376462330
25 IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_001    0.719115324
26 IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_002    0.100904662
27 IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_003    0.072124756
28 IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_004    0.001791312
29   IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_001    0.853820598
30   IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_002    0.040843806
31   IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_003    0.103557312
32   IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_004    0.124691019
33 IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_001    0.364191251
34 IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_002    0.115577889
35 IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_003    0.724804178
36 IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_004    0.360289855
37 IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_001    0.306001691
38 IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_002    0.000000000
39 IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_003    0.642433234
   number_cells
1          3304
2          2346
3          2457
4          3625
5          3291
6          3845
7          2805
8          2809
9          1901
10         1202
11         1970
12         1746
13         2136
14         1862
15         1933
16         2777
17         3372
18         3048
19         3719
20         3887
21         3004
22         2589
23         2248
24         4274
25         3165
26         2874
27         3078
28         4466
29         2709
30         2228
31         2530
32         3641
33         1966
34         1393
35         1915
36         3450
37         2366
38         2532
39         2696</code></pre>
<p>We will check how balanced the classes are across the images.</p>
<pre class="r"><code># Total cells per class
as_tibble(colData(sce)) %&gt;%
  group_by(cell_labels) %&gt;%
  summarise(number_cells = n()) %&gt;%
  as.data.frame()</code></pre>
<pre><code>   cell_labels number_cells
1       B_cell         2779
2        BnCD4         2293
3        BnCD8         1616
4         CD38         1554
5          CD4         1167
6          CD8         1873
7           DC          810
8        HLADR         1443
9   Macrophage         1430
10  Neutrophil         1276
11     NK_cell          246
12         pDC          312
13      Stroma         1440
14        Treg          155
15       Tumor        10061
16  unlabelled        78704</code></pre>
<pre class="r"><code># Total cells per class and image
as_tibble(colData(sce)) %&gt;%
  filter(cell_labels != &quot;unlabelled&quot;) %&gt;%
  group_by(cell_labels, sample_id) %&gt;%
  summarise(number_cells = n()) %&gt;%
  DT::datatable()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;cell_labels&#39;. You can override using the
`.groups` argument.</code></pre>
<div id="htmlwidget-9a2ba10e4c840f8b613b" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-9a2ba10e4c840f8b613b">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120"],["B_cell","B_cell","B_cell","B_cell","B_cell","B_cell","B_cell","B_cell","BnCD4","BnCD4","BnCD4","BnCD4","BnCD4","BnCD4","BnCD4","BnCD4","BnCD8","BnCD8","BnCD8","BnCD8","BnCD8","BnCD8","BnCD8","BnCD8","CD38","CD38","CD38","CD38","CD38","CD38","CD38","CD38","CD4","CD4","CD4","CD4","CD4","CD4","CD4","CD4","CD8","CD8","CD8","CD8","CD8","CD8","CD8","CD8","DC","DC","DC","DC","DC","DC","DC","DC","HLADR","HLADR","HLADR","HLADR","HLADR","HLADR","HLADR","HLADR","Macrophage","Macrophage","Macrophage","Macrophage","Macrophage","Macrophage","Macrophage","Macrophage","Neutrophil","Neutrophil","Neutrophil","Neutrophil","Neutrophil","Neutrophil","Neutrophil","Neutrophil","NK_cell","NK_cell","NK_cell","NK_cell","NK_cell","NK_cell","NK_cell","NK_cell","pDC","pDC","pDC","pDC","pDC","pDC","pDC","pDC","Stroma","Stroma","Stroma","Stroma","Stroma","Stroma","Stroma","Stroma","Treg","Treg","Treg","Treg","Treg","Treg","Treg","Tumor","Tumor","Tumor","Tumor","Tumor","Tumor","Tumor","Tumor","Tumor"],["IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_004","IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_008","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_008","IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_004","IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_004","IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_004","IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_008","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_008","IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_004","IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_004","IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_004","IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_008","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_008","IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_004","IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_004","IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_005","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_005","IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_003","IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_006","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_006","IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_001","IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_006","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_006","IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_001","IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_004","IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_006","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_008","IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_001","IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_004","IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_006","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_008","IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_001","IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_004","IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_006","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_008","IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_001","IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_004","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_004","IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_005","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_008","IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_002","IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_006","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_005","IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_001","IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_007","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_006","IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_004","IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_001","IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_007","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_005","IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_001","IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_006","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_006","IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_001","IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_005","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_007","IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_001","IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_003"],[724,294,489,414,94,66,496,202,437,363,469,555,43,120,215,91,185,187,108,129,153,268,532,54,50,258,53,603,244,16,262,68,141,66,74,253,77,374,120,62,268,77,251,84,918,125,35,115,37,40,261,95,90,110,104,73,198,115,426,168,116,153,102,165,255,145,314,149,123,326,112,6,338,20,451,99,115,91,117,45,46,50,11,85,7,3,23,21,9,2,261,8,8,2,3,19,68,180,74,323,100,100,138,457,29,41,3,11,11,55,5,1207,802,1183,1493,813,925,1109,1388,1141]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>cell_labels<\/th>\n      <th>sample_id<\/th>\n      <th>number_cells<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="visualize-markers-and-cells" class="section level2">
<h2>Visualize markers and cells</h2>
<p>Now, we will check the expression of selected markers across the
classes and visualize cell labels on tSNE.</p>
<pre class="r"><code>library(scran)</code></pre>
<pre><code>Loading required package: scuttle</code></pre>
<pre class="r"><code>library(dittoSeq)</code></pre>
<pre><code>Loading required package: ggplot2</code></pre>
<pre class="r"><code>library(viridis)</code></pre>
<pre><code>Loading required package: viridisLite</code></pre>
<pre class="r"><code>lab_sce &lt;- sce[,sce$cell_labels != &quot;unlabelled&quot;]
agr_sce &lt;- aggregateAcrossCells(lab_sce, 
                                ids = colData(lab_sce)[,c(&quot;sample_id&quot;, &quot;cell_labels&quot;)], 
                                statistics = &quot;mean&quot;)
assay(agr_sce, &quot;exprs&quot;) &lt;- asinh(counts(agr_sce))
colnames(agr_sce) &lt;- paste0(agr_sce$sample_id, &quot;_&quot;, 
                            agr_sce$cell_labels)

# Define markers that were used for gating
cur_markers &lt;- c(&quot;Ecad&quot;, &quot;CarbonicAnhydrase&quot;, &quot;Ki67&quot;, &quot;CD14&quot;, &quot;HLADR&quot;, 
                 &quot;CD11c&quot;, &quot;CD163&quot;, &quot;CD303&quot;, &quot;CD68&quot;, &quot;CD15&quot;, &quot;MPO&quot;, &quot;CD38&quot;,
                 &quot;SMA&quot;, &quot;PDGFRb&quot;, &quot;CD20&quot;, &quot;CD7&quot;, &quot;CD3&quot;, &quot;CD8a&quot;, &quot;CD4&quot;, &quot;GrzB&quot;, &quot;FOXP3&quot;, &quot;PD1&quot;)

# Non-scaled
dittoHeatmap(agr_sce[cur_markers,], assay = &quot;exprs&quot;,
            annot.by = c(&quot;patient_id&quot;, &quot;cell_labels&quot;, &quot;indication&quot;), 
            cluster_rows = FALSE,
            scale = &quot;none&quot;, heatmap.colors = viridis(100), order.by = &quot;cell_labels&quot;,
            annotation_colors = list(patient_id = metadata(sce)$colour_vectors$PatientIds,
                                     cell_labels = metadata(sce)$colour_vectors$cell_types,
                                     indication = metadata(sce)$colour_vectors$Indication))</code></pre>
<p><img src="figure/05_labelling.Rmd/quality-control-4-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-quality-control-4-1">
Past versions of quality-control-4-1.png
</button>
</p>
<div id="fig-quality-control-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/8d1f8ec230505f81891267366712aeb5847bda1a/IMC/Analysis/docs/figure/05_labelling.Rmd/quality-control-4-1.png" target="_blank">8d1f8ec</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/f7cc35b2360b342abaa00a75373b5f8a03f47633/IMC/Analysis/docs/figure/05_labelling.Rmd/quality-control-4-1.png" target="_blank">f7cc35b</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Centered and scaled
dittoHeatmap(agr_sce[cur_markers,], assay = &quot;exprs&quot;,
            annot.by = c(&quot;patient_id&quot;, &quot;cell_labels&quot;, &quot;indication&quot;), 
            cluster_rows = FALSE, order.by = &quot;cell_labels&quot;,
            annotation_colors = list(patient_id = metadata(sce)$colour_vectors$PatientIds,
                                     cell_labels = metadata(sce)$colour_vectors$cell_types,
                                     indication = metadata(sce)$colour_vectors$Indication),
            heatmap.colors = colorRampPalette(c(&quot;dark blue&quot;, &quot;white&quot;, &quot;dark red&quot;))(100),
            breaks = seq(-3, 3, length.out = 101))</code></pre>
<p><img src="figure/05_labelling.Rmd/quality-control-4-2.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-quality-control-4-2">
Past versions of quality-control-4-2.png
</button>
</p>
<div id="fig-quality-control-4-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/8d1f8ec230505f81891267366712aeb5847bda1a/IMC/Analysis/docs/figure/05_labelling.Rmd/quality-control-4-2.png" target="_blank">8d1f8ec</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/f7cc35b2360b342abaa00a75373b5f8a03f47633/IMC/Analysis/docs/figure/05_labelling.Rmd/quality-control-4-2.png" target="_blank">f7cc35b</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Finally, we will represent the gated cells on tSNE.</p>
<pre class="r"><code>cur_unlab_sce &lt;- sce[,sce$cell_labels == &quot;unlabelled&quot;]
cur_lab_sce &lt;- sce[,sce$cell_labels != &quot;unlabelled&quot;]

ggplot() +
    geom_point(aes(x = UMAP_1, y = UMAP_2, colour = cell_labels), 
              data = data.frame(UMAP_1 = reducedDim(cur_unlab_sce, &quot;UMAP_fastMNN&quot;)[,1],
                                UMAP_2 = reducedDim(cur_unlab_sce, &quot;UMAP_fastMNN&quot;)[,2],
                                cell_labels = colData(cur_unlab_sce)$cell_labels)) +
    geom_point(aes(x = UMAP_1, y = UMAP_2, colour = cell_labels), size = 0.5, 
            data = data.frame(UMAP_1 = reducedDim(cur_lab_sce, &quot;UMAP_fastMNN&quot;)[,1],
                              UMAP_2 = reducedDim(cur_lab_sce, &quot;UMAP_fastMNN&quot;)[,2],
                              cell_labels = colData(cur_lab_sce)$cell_labels)) +
    scale_color_manual(values = metadata(sce)$colour_vectors$cell_types) + 
    theme_bw()</code></pre>
<p><img src="figure/05_labelling.Rmd/tsne-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-tsne-1">
Past versions of tsne-1.png
</button>
</p>
<div id="fig-tsne-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/8d1f8ec230505f81891267366712aeb5847bda1a/IMC/Analysis/docs/figure/05_labelling.Rmd/tsne-1.png" target="_blank">8d1f8ec</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/f7cc35b2360b342abaa00a75373b5f8a03f47633/IMC/Analysis/docs/figure/05_labelling.Rmd/tsne-1.png" target="_blank">f7cc35b</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="save-sce-object" class="section level1">
<h1>Save SCE object</h1>
<pre class="r"><code>saveRDS(sce, &quot;/Volumes/immucan_volume/processed_data/Panel_1/2022_WORKFLOW/IMC/Rout/sce.rds&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] viridis_0.6.2               viridisLite_0.4.0          
 [3] dittoSeq_1.9.1              ggplot2_3.3.6              
 [5] scran_1.25.0                scuttle_1.7.2              
 [7] dplyr_1.0.9                 stringr_1.4.0              
 [9] cytomapper_1.9.0            SingleCellExperiment_1.19.0
[11] SummarizedExperiment_1.27.1 Biobase_2.57.1             
[13] GenomicRanges_1.49.0        GenomeInfoDb_1.33.3        
[15] IRanges_2.31.0              S4Vectors_0.35.1           
[17] BiocGenerics_0.43.0         MatrixGenerics_1.9.0       
[19] matrixStats_0.62.0          EBImage_4.39.0             
[21] workflowr_1.7.0            

loaded via a namespace (and not attached):
  [1] systemfonts_1.0.4         plyr_1.8.7               
  [3] igraph_1.3.1              sp_1.5-0                 
  [5] shinydashboard_0.7.2      BiocParallel_1.31.8      
  [7] crosstalk_1.2.0           digest_0.6.29            
  [9] htmltools_0.5.2           tiff_0.1-11              
 [11] fansi_1.0.3               magrittr_2.0.3           
 [13] ScaledMatrix_1.5.0        cluster_2.1.3            
 [15] limma_3.53.3              svgPanZoom_0.3.4         
 [17] svglite_2.1.0             jpeg_0.1-9               
 [19] colorspace_2.0-3          ggrepel_0.9.1            
 [21] xfun_0.31                 callr_3.7.0              
 [23] crayon_1.5.1              RCurl_1.98-1.7           
 [25] jsonlite_1.8.0            glue_1.6.2               
 [27] gtable_0.3.0              nnls_1.4                 
 [29] zlibbioc_1.43.0           XVector_0.37.0           
 [31] DelayedArray_0.23.0       BiocSingular_1.13.0      
 [33] Rhdf5lib_1.19.2           HDF5Array_1.25.1         
 [35] abind_1.4-5               scales_1.2.0             
 [37] pheatmap_1.0.12           DBI_1.1.2                
 [39] edgeR_3.39.1              Rcpp_1.0.8.3             
 [41] xtable_1.8-4              dqrng_0.3.0              
 [43] rsvd_1.0.5                DT_0.23                  
 [45] metapod_1.5.0             htmlwidgets_1.5.4        
 [47] httr_1.4.3                RColorBrewer_1.1-3       
 [49] ellipsis_0.3.2            farver_2.1.0             
 [51] pkgconfig_2.0.3           sass_0.4.1               
 [53] locfit_1.5-9.5            utf8_1.2.2               
 [55] labeling_0.4.2            tidyselect_1.1.2         
 [57] rlang_1.0.2               later_1.3.0              
 [59] munsell_0.5.0             tools_4.2.0              
 [61] cli_3.3.0                 generics_0.1.2           
 [63] ggridges_0.5.3            evaluate_0.15            
 [65] fastmap_1.1.0             fftwtools_0.9-11         
 [67] yaml_2.3.5                processx_3.6.0           
 [69] knitr_1.39                fs_1.5.2                 
 [71] purrr_0.3.4               sparseMatrixStats_1.9.0  
 [73] whisker_0.4               mime_0.12                
 [75] compiler_4.2.0            rstudioapi_0.13          
 [77] beeswarm_0.4.0            png_0.1-7                
 [79] tibble_3.1.7              statmod_1.4.36           
 [81] bslib_0.3.1               stringi_1.7.6            
 [83] highr_0.9                 ps_1.7.0                 
 [85] lattice_0.20-45           bluster_1.7.0            
 [87] Matrix_1.4-1              vctrs_0.4.1              
 [89] pillar_1.7.0              lifecycle_1.0.1          
 [91] rhdf5filters_1.9.0        jquerylib_0.1.4          
 [93] BiocNeighbors_1.15.1      cowplot_1.1.1            
 [95] bitops_1.0-7              irlba_2.3.5              
 [97] raster_3.5-15             httpuv_1.6.5             
 [99] R6_2.5.1                  promises_1.2.0.1         
[101] gridExtra_2.3             vipor_0.4.5              
[103] codetools_0.2-18          assertthat_0.2.1         
[105] rhdf5_2.41.0              rprojroot_2.0.3          
[107] withr_2.5.0               GenomeInfoDbData_1.2.8   
[109] parallel_4.2.0            terra_1.5-21             
[111] grid_4.2.0                beachmat_2.13.3          
[113] rmarkdown_2.14            DelayedMatrixStats_1.19.0
[115] git2r_0.30.1              getPass_0.2-2            
[117] shiny_1.7.1               ggbeeswarm_0.6.0         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
