<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Label cell types</title>

<script src="site_libs/header-attrs-2.17/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.26/datatables.js"></script>
<link href="site_libs/dt-core-1.12.1/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.12.1/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.12.1/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">IMMUcanWorkflow2022</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Label cell types</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-11-09
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220321code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220321)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220321code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220321)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomImmucanWP7IMMUcanWorkflow2021treeebe28aeb85ab68f57f66ce1618fe58f96a5d35a8targetblankebe28aea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/tree/ebe28aeb85ab68f57f66ce1618fe58f96a5d35a8" target="_blank">ebe28ae</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomImmucanWP7IMMUcanWorkflow2021treeebe28aeb85ab68f57f66ce1618fe58f96a5d35a8targetblankebe28aea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/tree/ebe28aeb85ab68f57f66ce1618fe58f96a5d35a8" target="_blank">ebe28ae</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    Alignment/
    Ignored:    IMC/.DS_Store
    Ignored:    IMC/Analysis/.DS_Store
    Ignored:    IMC/Analysis/.Rproj.user/
    Ignored:    comparisons/.Rproj.user/
    Ignored:    comparisons/code/.ipynb_checkpoints/
    Ignored:    mIF/.DS_Store
    Ignored:    mIF/Analysis/.Rproj.user/

Unstaged changes:
    Modified:   IMC/Analysis/analysis/07_predict_celltypes.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>IMC/Analysis/analysis/05_labelling.Rmd</code>) and HTML
(<code>IMC/Analysis/docs/05_labelling.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/c28e2279f4ba44f0b4c265f5133bfaf10078372b/IMC/Analysis/analysis/05_labelling.Rmd" target="_blank">c28e227</a>
</td>
<td>
nilseling
</td>
<td>
2022-10-21
</td>
<td>
Final classifier
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/e402ead09f7fcf1c50cd3623df965c7764bcd389/IMC/Analysis/analysis/05_labelling.Rmd" target="_blank">e402ead</a>
</td>
<td>
nilseling
</td>
<td>
2022-10-13
</td>
<td>
Adjustments for new samples
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ImmucanWP7/IMMUcanWorkflow2021/329cfe69b24cf8a10aa69b2fb4a27b155bfcc091/IMC/Analysis/docs/05_labelling.html" target="_blank">329cfe6</a>
</td>
<td>
nilseling
</td>
<td>
2022-08-19
</td>
<td>
Added phenotype key for mIF
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/4554a259201ef6a52df476c8208dede506b41b04/IMC/Analysis/analysis/05_labelling.Rmd" target="_blank">4554a25</a>
</td>
<td>
nilseling
</td>
<td>
2022-06-24
</td>
<td>
Recompiled scripts
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ImmucanWP7/IMMUcanWorkflow2021/4554a259201ef6a52df476c8208dede506b41b04/IMC/Analysis/docs/05_labelling.html" target="_blank">4554a25</a>
</td>
<td>
nilseling
</td>
<td>
2022-06-24
</td>
<td>
Recompiled scripts
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/8d1f8ec230505f81891267366712aeb5847bda1a/IMC/Analysis/analysis/05_labelling.Rmd" target="_blank">8d1f8ec</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-30
</td>
<td>
Recompiled workflow
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ImmucanWP7/IMMUcanWorkflow2021/8d1f8ec230505f81891267366712aeb5847bda1a/IMC/Analysis/docs/05_labelling.html" target="_blank">8d1f8ec</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-30
</td>
<td>
Recompiled workflow
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ImmucanWP7/IMMUcanWorkflow2021/f7cc35b2360b342abaa00a75373b5f8a03f47633/IMC/Analysis/docs/05_labelling.html" target="_blank">f7cc35b</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-11
</td>
<td>
Recompiled scripts
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/c22d4e1fe74fdd577574bf3c0c514bb5df0a06fa/IMC/Analysis/analysis/05_labelling.Rmd" target="_blank">c22d4e1</a>
</td>
<td>
nilseling
</td>
<td>
2022-04-29
</td>
<td>
Added more projects
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Here, we will manually label cells based on their cell type. This is
done based on the labelling scheme in
<code>data/UZH_Gating_scheme_final.pptx</code></p>
<div id="load-data" class="section level2">
<h2>Load data</h2>
<p>First, we will load the previously generated
<code>SingleCellExperiment</code> and <code>CytoImageList</code>
objects.</p>
<pre class="r"><code>library(cytomapper)
library(stringr)

sce &lt;- readRDS(&quot;/Volumes/immucan_volume/processed_data/Panel_1/2022_WORKFLOW/IMC/Rout/sce.rds&quot;)
images &lt;- readRDS(&quot;/Volumes/immucan_volume/processed_data/Panel_1/2022_WORKFLOW/IMC/Rout/images.rds&quot;)
masks &lt;- readRDS(&quot;/Volumes/immucan_volume/processed_data/Panel_1/2022_WORKFLOW/IMC/Rout/masks.rds&quot;)</code></pre>
</div>
<div id="label-images" class="section level2">
<h2>Label images</h2>
<p>We will now label cells on images and save them.</p>
<pre class="r"><code>images &lt;- CytoImageList(images, on_disk = FALSE)
masks &lt;- CytoImageList(masks, on_disk = FALSE)

# Change display name
sce$ID &lt;- paste0(sce$indication, &quot;_&quot;, sce$patient_id, &quot;_&quot;, sce$ROI)
cur_df &lt;- colData(sce)[,c(&quot;sample_id&quot;, &quot;ID&quot;)]
cur_mcols &lt;- mcols(images)

cur_mcols &lt;- merge(cur_mcols, cur_df, by = &quot;sample_id&quot;, sort = FALSE)
cur_mcols &lt;- unique(cur_mcols)

all.equal(cur_mcols$sample_id, mcols(images)$sample_id)</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code>mcols(images) &lt;- mcols(masks) &lt;- cur_mcols 

if (interactive()) {
  cytomapperShiny(object = sce, mask = masks, image = images,
                  cell_id = &quot;ObjectNumber&quot;, img_id = &quot;ID&quot;)
}</code></pre>
</div>
<div id="quality-control" class="section level2">
<h2>Quality control</h2>
<p>We next check for labelling quality.</p>
<pre class="r"><code>library(stringr)
library(SingleCellExperiment)
label.files &lt;- list.files(&quot;/Volumes/immucan_volume/processed_data/Panel_1/2022_WORKFLOW/IMC/Rout/gated_cells/&quot;, 
                          full.names = TRUE, pattern = &quot;.rds$&quot;)

# Read in SCE objects
cur_sces &lt;- lapply(label.files, readRDS)

# Merge SCE objects
cur_names &lt;- Reduce(intersect, x = lapply(cur_sces, function(x){colnames(colData(x))}))
cur_sces &lt;- lapply(cur_sces, function(x){
  colData(x) &lt;- colData(x)[,cur_names]
  return(x)
})
cur_sce &lt;- do.call(&quot;cbind&quot;, cur_sces)

# now we check if the celltypes were labelled uniquely
cur_sce$cytomapper_CellLabel &lt;- str_split(cur_sce$cytomapper_CellLabel, &quot;_&quot;, simplify = TRUE)[,1]
names &lt;- unique(cur_sce$cytomapper_CellLabel)
unique(names)</code></pre>
<pre><code> [1] &quot;Bcell&quot;      &quot;BnTcell&quot;    &quot;CD4&quot;        &quot;CD8&quot;        &quot;DC&quot;        
 [6] &quot;HLADR&quot;      &quot;MacCD163&quot;   &quot;Mural&quot;      &quot;Neutrophil&quot; &quot;NK&quot;        
[11] &quot;pDC&quot;        &quot;Plasma&quot;     &quot;Treg&quot;       &quot;Tumor&quot;     </code></pre>
<pre class="r"><code>table(cur_sce$cytomapper_CellLabel, cur_sce$indication)</code></pre>
<pre><code>            
              BCC  CRC NSCLC  RCC SCCHN
  Bcell        77  149   152  269   172
  BnTcell     177  355   180  264   341
  CD4           6   68    41   51   215
  CD8          17   25   131   58   181
  DC           49  105   104  170    53
  HLADR       161  141   222  332    91
  MacCD163     82  102   315  169    65
  Mural       347  372    34  307   148
  Neutrophil   48    0    97  266   682
  NK            7    6     4   36    15
  pDC         173   83    16   27   159
  Plasma       58  106   187   17   148
  Treg         10   20    40    3   165
  Tumor       746 1194  2618 4037  2050</code></pre>
<p>We next need to remove cells that were assigned to different cell
types. For these cell, the classification uncertainty will be
highest.</p>
<pre class="r"><code>cur_vec &lt;- cur_sce$cytomapper_CellLabel

cur_tab &lt;- unclass(table(colnames(cur_sce), cur_sce$cytomapper_CellLabel))
cur_labels &lt;- rep(&quot;doublets&quot;, nrow(cur_tab))
names(cur_labels) &lt;- rownames(cur_tab)

# Single assignments
single_index &lt;- rowSums(cur_tab) == 1
cur_labels[single_index] &lt;- colnames(cur_tab)[apply(cur_tab[single_index,], 1, 
                                                    which.max)]

# Double assignment within the tumor
double_index &lt;- rowSums(cur_tab) == 2 &amp; cur_tab[,&quot;Tumor&quot;] == 1
no_tumor &lt;- cur_tab[,colnames(cur_tab) != &quot;Tumor&quot;]
cur_labels[double_index] &lt;- colnames(no_tumor)[apply(no_tumor[double_index,], 1, 
                                                    which.max)]

# Remove doublets
cur_labels &lt;- cur_labels[cur_labels != &quot;doublets&quot;]
table(cur_labels)</code></pre>
<pre><code>cur_labels
     Bcell    BnTcell        CD4        CD8         DC      HLADR   MacCD163 
       808       1314        379        412        473        937        721 
     Mural Neutrophil         NK        pDC     Plasma       Treg      Tumor 
      1208       1050         66        458        515        238      10641 </code></pre>
<pre class="r"><code># Store in SingleCellExperiment
sce_labels &lt;- rep(&quot;unlabelled&quot;, ncol(sce))
names(sce_labels) &lt;- colnames(sce)
sce_labels[names(cur_labels)] &lt;- cur_labels
sce$cell_labels &lt;- sce_labels

table(sce$cell_labels, sce$indication)</code></pre>
<pre><code>            
               BCC   CRC NSCLC   RCC SCCHN
  Bcell         75   149   152   260   172
  BnTcell      175   355   180   263   341
  CD4            6    67    41    51   214
  CD8           17    25   131    58   181
  DC            49   105   104   162    53
  HLADR        161   141   222   322    91
  MacCD163      82   102   315   157    65
  Mural        347   372    34   307   148
  Neutrophil    44     0    97   228   681
  NK             7     6     4    34    15
  pDC          173    83    16    27   159
  Plasma        58   105   187    17   148
  Treg          10    20    40     3   165
  Tumor        746  1194  2618  4033  2050
  unlabelled 13580 20840 23924 14766 16734</code></pre>
</div>
<div id="define-colour-vector" class="section level2">
<h2>Define colour vector</h2>
<p>In this step, we will define a colour vector for all further
visualization purposes.</p>
<pre class="r"><code>cell_types &lt;- c(&quot;Bcell&quot;, &quot;Plasma&quot;, &quot;CD4&quot;, &quot;CD8&quot;, &quot;Treg&quot;, &quot;BnTcell&quot;,
                &quot;DC&quot;, &quot;HLADR&quot;, &quot;MacCD163&quot;, &quot;Neutrophil&quot;, &quot;NK&quot;, &quot;pDC&quot;, &quot;Mural&quot;, 
                &quot;Tumor&quot;, &quot;unlabelled&quot;, &quot;undefined&quot;)

names(cell_types) &lt;- cell_types
cell_types[&quot;Bcell&quot;] &lt;- &quot;red3&quot;
cell_types[&quot;Plasma&quot;] &lt;- &quot;goldenrod2&quot;
cell_types[&quot;CD4&quot;] &lt;- &quot;palevioletred1&quot;
cell_types[&quot;CD8&quot;] &lt;- &quot;darkorchid&quot;
cell_types[&quot;Treg&quot;] &lt;- &quot;maroon4&quot;
cell_types[&quot;BnTcell&quot;] &lt;- &quot;darkorange4&quot;
cell_types[&quot;DC&quot;] &lt;- &quot;green4&quot;
cell_types[&quot;HLADR&quot;] &lt;- &quot;darkgreen&quot;
cell_types[&quot;MacCD163&quot;] &lt;- &quot;green&quot;
cell_types[&quot;Neutrophil&quot;] &lt;- &quot;blue1&quot;
cell_types[&quot;NK&quot;] &lt;- &quot;yellow&quot;
cell_types[&quot;pDC&quot;] &lt;- &quot;deepskyblue&quot;
cell_types[&quot;Mural&quot;] &lt;- &quot;tomato&quot;
cell_types[&quot;Tumor&quot;] &lt;- &quot;sienna4&quot;
cell_types[&quot;unlabelled&quot;] &lt;- &quot;gray&quot;
cell_types[&quot;undefined&quot;] &lt;- &quot;black&quot;

metadata(sce)$colour_vectors$cell_types &lt;- cell_types

pie(c(rep(1,length(metadata(sce)$colour_vectors$cell_types))),
    col = metadata(sce)$colour_vectors$cell_types,
    labels = names(metadata(sce)$colour_vectors$cell_types))</code></pre>
<p><img src="figure/05_labelling.Rmd/define-color-vector-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-define-color-vector-1">
Past versions of define-color-vector-1.png
</button>
</p>
<div id="fig-define-color-vector-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/f7cc35b2360b342abaa00a75373b5f8a03f47633/IMC/Analysis/docs/figure/05_labelling.Rmd/define-color-vector-1.png" target="_blank">f7cc35b</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="quality-control-of-labelling-results" class="section level2">
<h2>Quality control of labelling results</h2>
<p>Here, we will check a number of statistics to assess labelling
quality.</p>
<pre class="r"><code>library(dplyr)
# 1. How many cells of how many images are labelled

# Percent cells labelled
as_tibble(colData(sce)) %&gt;% 
  summarise(labelled_cells = sum(cell_labels != &quot;unlabelled&quot;)/n()) </code></pre>
<pre><code># A tibble: 1 × 1
  labelled_cells
           &lt;dbl&gt;
1          0.176</code></pre>
<pre class="r"><code># Percent images labelled
as_tibble(colData(sce)) %&gt;% 
  group_by(sample_id) %&gt;%
  summarise(labelled_cells = sum(cell_labels != &quot;unlabelled&quot;)) %&gt;%
  ungroup() %&gt;%
  summarise(labelled_images = sum(labelled_cells != 0)/n())</code></pre>
<pre><code># A tibble: 1 × 1
  labelled_images
            &lt;dbl&gt;
1           0.842</code></pre>
<pre class="r"><code># Percent patients labelled
as_tibble(colData(sce)) %&gt;% 
  group_by(patient_id) %&gt;%
  summarise(labelled_cells = sum(cell_labels != &quot;unlabelled&quot;)) %&gt;%
  ungroup() %&gt;%
  summarise(labelled_samples = sum(labelled_cells != 0)/n())</code></pre>
<pre><code># A tibble: 1 × 1
  labelled_samples
             &lt;dbl&gt;
1                1</code></pre>
<pre class="r"><code># Percent of cells labelled per image
as_tibble(colData(sce)) %&gt;% 
  group_by(sample_id) %&gt;%
  summarise(labelled_cells = sum(cell_labels != &quot;unlabelled&quot;)/n(),
            number_cells = n()) %&gt;%
  as.data.frame()</code></pre>
<pre><code>                                                    sample_id labelled_cells
1     IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_001    0.409104522
2     IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_002    0.200410572
3     IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_003    0.005996473
4     IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_004    0.470055710
5     IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_001    0.158230734
6     IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_002    0.207207207
7     IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_003    0.102370146
8     IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_004    0.104890149
9     IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_005    0.004670169
10    IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_006    0.000000000
11    IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_007    0.281449893
12    IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_008    0.137200504
13    IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_005    0.000000000
14    IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_006    0.446469248
15    IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_007    0.269745360
16    IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_008    0.302883506
17    IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_001    0.291355721
18    IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_002    0.183652531
19    IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_003    0.203492063
20    IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_004    0.090447827
21      IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_001    0.000000000
22      IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_002    0.215574131
23      IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_003    0.548312883
24      IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_004    0.022371968
25    IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_001    0.086086086
26    IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_002    0.004216444
27    IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_003    0.564499484
28    IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_004    0.027706370
29    IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_001    0.168664737
30    IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_002    0.605355064
31    IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_003    0.160058203
32   IMMUcan_Batch20210701_10075572-SPECT-VAR-TIS-UNST-03_001    0.000000000
33   IMMUcan_Batch20210701_10075572-SPECT-VAR-TIS-UNST-03_002    0.099860009
34   IMMUcan_Batch20210701_10075572-SPECT-VAR-TIS-UNST-03_003    0.000000000
35   IMMUcan_Batch20210701_10075572-SPECT-VAR-TIS-UNST-03_004    0.148898678
36 IMMUcan_Batch20210921_10082495-SPECT-VAR-TIS-01-IMC-01_001    0.000000000
37 IMMUcan_Batch20210921_10082495-SPECT-VAR-TIS-01-IMC-01_002    0.439024390
38 IMMUcan_Batch20210921_10082495-SPECT-VAR-TIS-01-IMC-01_004    0.045706737
   number_cells
1          3339
2          3897
3          2835
4          2872
5          2193
6          1887
7          1983
8          2822
9          3426
10         3099
11         3752
12         3965
13         3071
14         2634
15         2317
16         4335
17         3216
18         2924
19         3150
20         4533
21         2768
22         2273
23         2608
24         3710
25         1998
26         1423
27         1938
28         3501
29         2419
30         2577
31         2749
32         2829
33         2143
34         3828
35         3405
36         1630
37         1230
38         3785</code></pre>
<p>We will check how balanced the classes are across the images.</p>
<pre class="r"><code># Total cells per class
as_tibble(colData(sce)) %&gt;%
  group_by(cell_labels) %&gt;%
  summarise(number_cells = n()) %&gt;%
  as.data.frame()</code></pre>
<pre><code>   cell_labels number_cells
1        Bcell          808
2      BnTcell         1314
3          CD4          379
4          CD8          412
5           DC          473
6        HLADR          937
7     MacCD163          721
8        Mural         1208
9   Neutrophil         1050
10          NK           66
11         pDC          458
12      Plasma          515
13        Treg          238
14       Tumor        10641
15  unlabelled        89844</code></pre>
<pre class="r"><code># Total cells per class and image
as_tibble(colData(sce)) %&gt;%
  filter(cell_labels != &quot;unlabelled&quot;) %&gt;%
  group_by(cell_labels, sample_id) %&gt;%
  summarise(number_cells = n()) %&gt;%
  DT::datatable()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;cell_labels&#39;. You can override using the
`.groups` argument.</code></pre>
<div id="htmlwidget-9a2ba10e4c840f8b613b" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-9a2ba10e4c840f8b613b">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103"],["Bcell","Bcell","Bcell","Bcell","Bcell","Bcell","BnTcell","BnTcell","BnTcell","BnTcell","BnTcell","BnTcell","CD4","CD4","CD4","CD4","CD4","CD4","CD8","CD8","CD8","CD8","CD8","CD8","DC","DC","DC","DC","DC","DC","DC","HLADR","HLADR","HLADR","HLADR","HLADR","HLADR","HLADR","MacCD163","MacCD163","MacCD163","MacCD163","MacCD163","MacCD163","MacCD163","Mural","Mural","Mural","Mural","Mural","Mural","Mural","Mural","Neutrophil","Neutrophil","Neutrophil","Neutrophil","Neutrophil","Neutrophil","NK","NK","NK","NK","NK","NK","NK","NK","NK","NK","NK","pDC","pDC","pDC","pDC","pDC","pDC","Plasma","Plasma","Plasma","Plasma","Plasma","Plasma","Plasma","Plasma","Treg","Treg","Treg","Treg","Treg","Treg","Tumor","Tumor","Tumor","Tumor","Tumor","Tumor","Tumor","Tumor","Tumor","Tumor","Tumor","Tumor","Tumor"],["IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_004","IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_008","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_008","IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_Batch20210701_10075572-SPECT-VAR-TIS-UNST-03_004","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_004","IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_008","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_008","IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_Batch20210701_10075572-SPECT-VAR-TIS-UNST-03_004","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_004","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_008","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_008","IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_003","IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_004","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_008","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_008","IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_003","IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_008","IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_002","IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_Batch20210921_10082495-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_008","IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_002","IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_Batch20210921_10082495-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_008","IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_002","IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_Batch20210921_10082495-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_007","IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_004","IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_002","IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_003","IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_004","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_007","IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_004","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_004","IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_007","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_006","IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_002","IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_Batch20210701_10075572-SPECT-VAR-TIS-UNST-03_004","IMMUcan_Batch20210921_10082495-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_004","IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_005","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_008","IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_004","IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_Batch20210921_10082495-SPECT-VAR-TIS-01-IMC-01_004","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_007","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_008","IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_004","IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_003","IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_004","IMMUcan_Batch20210921_10082495-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_004","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_008","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_008","IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_003","IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10061074-SPECT-VAR-TIS-01-IMC-01_004","IMMUcan_2022_WFLOW_10067433-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_2022_WFLOW_10068868-SPECT-VAR-TIS-01-IMC-01_007","IMMUcan_2022_WFLOW_10071582-SPECT-VAR-TIS-01-IMC-01_006","IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_001","IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_2022_WFLOW_10073140-SPECT-VAR-TIS-01-IMC-01_004","IMMUcan_2022_WFLOW_10074349-SPECT-VAR-TIS-UNST-03_003","IMMUcan_2022_WFLOW_10074832-SPECT-VAR-TIS-01-IMC-01_003","IMMUcan_2022_WFLOW_10075371-SPECT-VAR-TIS-01-IMC-01_002","IMMUcan_Batch20210701_10075572-SPECT-VAR-TIS-UNST-03_002","IMMUcan_Batch20210921_10082495-SPECT-VAR-TIS-01-IMC-01_002"],[139,75,152,172,121,149,159,175,180,341,104,355,51,6,41,154,67,60,58,17,131,132,25,49,76,36,53,104,105,86,13,67,59,91,222,141,255,102,112,40,65,315,102,45,42,74,347,148,34,139,233,34,199,228,44,477,97,172,32,8,4,2,4,9,3,6,6,16,3,5,11,16,159,83,16,173,17,35,170,51,17,105,97,23,3,10,40,95,20,70,1284,1223,391,882,1167,937,440,359,980,883,1526,214,355]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>cell_labels<\/th>\n      <th>sample_id<\/th>\n      <th>number_cells<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="visualize-markers-and-cells" class="section level2">
<h2>Visualize markers and cells</h2>
<p>Now, we will check the expression of selected markers across the
classes and visualize cell labels on tSNE.</p>
<pre class="r"><code>library(scran)</code></pre>
<pre><code>Loading required package: scuttle</code></pre>
<pre class="r"><code>library(dittoSeq)</code></pre>
<pre><code>Loading required package: ggplot2</code></pre>
<pre class="r"><code>library(viridis)</code></pre>
<pre><code>Loading required package: viridisLite</code></pre>
<pre class="r"><code>lab_sce &lt;- sce[,sce$cell_labels != &quot;unlabelled&quot;]
agr_sce &lt;- aggregateAcrossCells(lab_sce, 
                                ids = colData(lab_sce)[,c(&quot;sample_id&quot;, &quot;cell_labels&quot;)], 
                                statistics = &quot;mean&quot;)
assay(agr_sce, &quot;exprs&quot;) &lt;- asinh(counts(agr_sce))
colnames(agr_sce) &lt;- paste0(agr_sce$sample_id, &quot;_&quot;, 
                            agr_sce$cell_labels)

# Define markers that were used for gating
cur_markers &lt;- c(&quot;Ecad&quot;, &quot;CarbonicAnhydrase&quot;, &quot;Ki67&quot;, &quot;CD14&quot;, &quot;HLADR&quot;, 
                 &quot;CD11c&quot;, &quot;CD163&quot;, &quot;CD303&quot;, &quot;CD68&quot;, &quot;CD15&quot;, &quot;MPO&quot;, &quot;CD38&quot;,
                 &quot;SMA&quot;, &quot;PDGFRb&quot;, &quot;CD20&quot;, &quot;CD7&quot;, &quot;CD3&quot;, &quot;CD8a&quot;, &quot;CD4&quot;, &quot;GrzB&quot;, &quot;FOXP3&quot;, &quot;PD1&quot;)

# Non-scaled
dittoHeatmap(agr_sce[cur_markers,], assay = &quot;exprs&quot;,
            annot.by = c(&quot;patient_id&quot;, &quot;cell_labels&quot;, &quot;indication&quot;), 
            cluster_rows = FALSE,
            scale = &quot;none&quot;, heatmap.colors = viridis(100), order.by = &quot;cell_labels&quot;,
            annotation_colors = list(patient_id = metadata(sce)$colour_vectors$PatientIds,
                                     cell_labels = metadata(sce)$colour_vectors$cell_types,
                                     indication = metadata(sce)$colour_vectors$Indication))</code></pre>
<p><img src="figure/05_labelling.Rmd/quality-control-4-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-quality-control-4-1">
Past versions of quality-control-4-1.png
</button>
</p>
<div id="fig-quality-control-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/4554a259201ef6a52df476c8208dede506b41b04/IMC/Analysis/docs/figure/05_labelling.Rmd/quality-control-4-1.png" target="_blank">4554a25</a>
</td>
<td>
nilseling
</td>
<td>
2022-06-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/8d1f8ec230505f81891267366712aeb5847bda1a/IMC/Analysis/docs/figure/05_labelling.Rmd/quality-control-4-1.png" target="_blank">8d1f8ec</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/f7cc35b2360b342abaa00a75373b5f8a03f47633/IMC/Analysis/docs/figure/05_labelling.Rmd/quality-control-4-1.png" target="_blank">f7cc35b</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Centered and scaled
dittoHeatmap(agr_sce[cur_markers,], assay = &quot;exprs&quot;,
            annot.by = c(&quot;patient_id&quot;, &quot;cell_labels&quot;, &quot;indication&quot;), 
            cluster_rows = FALSE, order.by = &quot;cell_labels&quot;,
            annotation_colors = list(patient_id = metadata(sce)$colour_vectors$PatientIds,
                                     cell_labels = metadata(sce)$colour_vectors$cell_types,
                                     indication = metadata(sce)$colour_vectors$Indication),
            heatmap.colors = colorRampPalette(c(&quot;dark blue&quot;, &quot;white&quot;, &quot;dark red&quot;))(100),
            breaks = seq(-3, 3, length.out = 101))</code></pre>
<p><img src="figure/05_labelling.Rmd/quality-control-4-2.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-quality-control-4-2">
Past versions of quality-control-4-2.png
</button>
</p>
<div id="fig-quality-control-4-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/4554a259201ef6a52df476c8208dede506b41b04/IMC/Analysis/docs/figure/05_labelling.Rmd/quality-control-4-2.png" target="_blank">4554a25</a>
</td>
<td>
nilseling
</td>
<td>
2022-06-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/8d1f8ec230505f81891267366712aeb5847bda1a/IMC/Analysis/docs/figure/05_labelling.Rmd/quality-control-4-2.png" target="_blank">8d1f8ec</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/f7cc35b2360b342abaa00a75373b5f8a03f47633/IMC/Analysis/docs/figure/05_labelling.Rmd/quality-control-4-2.png" target="_blank">f7cc35b</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Finally, we will represent the gated cells on tSNE.</p>
<pre class="r"><code>cur_unlab_sce &lt;- sce[,sce$cell_labels == &quot;unlabelled&quot;]
cur_lab_sce &lt;- sce[,sce$cell_labels != &quot;unlabelled&quot;]

ggplot() +
    geom_point(aes(x = UMAP_1, y = UMAP_2, colour = cell_labels), 
              data = data.frame(UMAP_1 = reducedDim(cur_unlab_sce, &quot;UMAP_fastMNN&quot;)[,1],
                                UMAP_2 = reducedDim(cur_unlab_sce, &quot;UMAP_fastMNN&quot;)[,2],
                                cell_labels = colData(cur_unlab_sce)$cell_labels)) +
    geom_point(aes(x = UMAP_1, y = UMAP_2, colour = cell_labels), size = 0.5, 
            data = data.frame(UMAP_1 = reducedDim(cur_lab_sce, &quot;UMAP_fastMNN&quot;)[,1],
                              UMAP_2 = reducedDim(cur_lab_sce, &quot;UMAP_fastMNN&quot;)[,2],
                              cell_labels = colData(cur_lab_sce)$cell_labels)) +
    scale_color_manual(values = metadata(sce)$colour_vectors$cell_types) + 
    theme_bw()</code></pre>
<p><img src="figure/05_labelling.Rmd/tsne-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-tsne-1">
Past versions of tsne-1.png
</button>
</p>
<div id="fig-tsne-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/4554a259201ef6a52df476c8208dede506b41b04/IMC/Analysis/docs/figure/05_labelling.Rmd/tsne-1.png" target="_blank">4554a25</a>
</td>
<td>
nilseling
</td>
<td>
2022-06-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/8d1f8ec230505f81891267366712aeb5847bda1a/IMC/Analysis/docs/figure/05_labelling.Rmd/tsne-1.png" target="_blank">8d1f8ec</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/f7cc35b2360b342abaa00a75373b5f8a03f47633/IMC/Analysis/docs/figure/05_labelling.Rmd/tsne-1.png" target="_blank">f7cc35b</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="save-sce-object" class="section level1">
<h1>Save SCE object</h1>
<pre class="r"><code>saveRDS(sce, &quot;/Volumes/immucan_volume/processed_data/Panel_1/2022_WORKFLOW/IMC/Rout/sce.rds&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] viridis_0.6.2               viridisLite_0.4.1          
 [3] dittoSeq_1.10.0             ggplot2_3.4.0              
 [5] scran_1.26.0                scuttle_1.8.0              
 [7] dplyr_1.0.10                stringr_1.4.1              
 [9] cytomapper_1.10.0           SingleCellExperiment_1.20.0
[11] SummarizedExperiment_1.28.0 Biobase_2.58.0             
[13] GenomicRanges_1.50.0        GenomeInfoDb_1.34.1        
[15] IRanges_2.32.0              S4Vectors_0.36.0           
[17] BiocGenerics_0.44.0         MatrixGenerics_1.10.0      
[19] matrixStats_0.62.0          EBImage_4.40.0             
[21] workflowr_1.7.0            

loaded via a namespace (and not attached):
  [1] systemfonts_1.0.4         igraph_1.3.5             
  [3] sp_1.5-1                  shinydashboard_0.7.2     
  [5] BiocParallel_1.32.0       crosstalk_1.2.0          
  [7] digest_0.6.30             htmltools_0.5.3          
  [9] magick_2.7.3              tiff_0.1-11              
 [11] fansi_1.0.3               magrittr_2.0.3           
 [13] ScaledMatrix_1.6.0        SpatialExperiment_1.8.0  
 [15] cluster_2.1.4             limma_3.54.0             
 [17] svgPanZoom_0.3.4          R.utils_2.12.1           
 [19] svglite_2.1.0             jpeg_0.1-9               
 [21] colorspace_2.0-3          ggrepel_0.9.2            
 [23] xfun_0.34                 callr_3.7.3              
 [25] RCurl_1.98-1.9            jsonlite_1.8.3           
 [27] glue_1.6.2                gtable_0.3.1             
 [29] nnls_1.4                  zlibbioc_1.44.0          
 [31] XVector_0.38.0            DelayedArray_0.24.0      
 [33] BiocSingular_1.14.0       DropletUtils_1.18.0      
 [35] Rhdf5lib_1.20.0           HDF5Array_1.26.0         
 [37] abind_1.4-5               scales_1.2.1             
 [39] pheatmap_1.0.12           DBI_1.1.3                
 [41] edgeR_3.40.0              Rcpp_1.0.9               
 [43] xtable_1.8-4              dqrng_0.3.0              
 [45] rsvd_1.0.5                DT_0.26                  
 [47] metapod_1.6.0             htmlwidgets_1.5.4        
 [49] httr_1.4.4                RColorBrewer_1.1-3       
 [51] ellipsis_0.3.2            farver_2.1.1             
 [53] pkgconfig_2.0.3           R.methodsS3_1.8.2        
 [55] sass_0.4.2                locfit_1.5-9.6           
 [57] utf8_1.2.2                labeling_0.4.2           
 [59] tidyselect_1.2.0          rlang_1.0.6              
 [61] later_1.3.0               munsell_0.5.0            
 [63] tools_4.2.1               cachem_1.0.6             
 [65] cli_3.4.1                 generics_0.1.3           
 [67] ggridges_0.5.4            evaluate_0.18            
 [69] fastmap_1.1.0             fftwtools_0.9-11         
 [71] yaml_2.3.6                processx_3.8.0           
 [73] knitr_1.40                fs_1.5.2                 
 [75] sparseMatrixStats_1.10.0  whisker_0.4              
 [77] mime_0.12                 R.oo_1.25.0              
 [79] compiler_4.2.1            rstudioapi_0.14          
 [81] beeswarm_0.4.0            png_0.1-7                
 [83] statmod_1.4.37            tibble_3.1.8             
 [85] bslib_0.4.1               stringi_1.7.8            
 [87] highr_0.9                 ps_1.7.2                 
 [89] lattice_0.20-45           bluster_1.8.0            
 [91] Matrix_1.5-1              vctrs_0.5.0              
 [93] pillar_1.8.1              lifecycle_1.0.3          
 [95] rhdf5filters_1.10.0       jquerylib_0.1.4          
 [97] BiocNeighbors_1.16.0      cowplot_1.1.1            
 [99] bitops_1.0-7              irlba_2.3.5.1            
[101] raster_3.6-3              httpuv_1.6.6             
[103] R6_2.5.1                  promises_1.2.0.1         
[105] gridExtra_2.3             vipor_0.4.5              
[107] codetools_0.2-18          assertthat_0.2.1         
[109] rhdf5_2.42.0              rprojroot_2.0.3          
[111] rjson_0.2.21              withr_2.5.0              
[113] GenomeInfoDbData_1.2.9    parallel_4.2.1           
[115] terra_1.6-17              grid_4.2.1               
[117] beachmat_2.14.0           rmarkdown_2.17           
[119] DelayedMatrixStats_1.20.0 git2r_0.30.1             
[121] getPass_0.2-2             shiny_1.7.3              
[123] ggbeeswarm_0.6.0         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
