<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Predict cell types</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">IMMUcanWorkflow2022 - IMC analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="01_read_data.html">Read data</a>
</li>
<li>
  <a href="02_spillover_correction.html">Spillover correction</a>
</li>
<li>
  <a href="03_quality_control.html">Quality control</a>
</li>
<li>
  <a href="04_sample_correction.html">Sample correction</a>
</li>
<li>
  <a href="05_predict_celltypes.html">Celltype prediction</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Predict cell types</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-02-16
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>Analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220321code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220321)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220321code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220321)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomImmucanWP7IMMUcanWorkflow2021tree7371b3e7ee27ab053d1f819996c1ce7624ca0948targetblank7371b3ea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/tree/7371b3e7ee27ab053d1f819996c1ce7624ca0948" target="_blank">7371b3e</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomImmucanWP7IMMUcanWorkflow2021tree7371b3e7ee27ab053d1f819996c1ce7624ca0948targetblank7371b3ea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/tree/7371b3e7ee27ab053d1f819996c1ce7624ca0948" target="_blank">7371b3e</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    Alignment/
    Ignored:    Figures/.Rproj.user/
    Ignored:    Figures/output/.DS_Store
    Ignored:    IMC/.DS_Store
    Ignored:    IMC/Analysis/.DS_Store
    Ignored:    IMC/Analysis/.Rproj.user/
    Ignored:    IMC/Analysis/analysis/figure/
    Ignored:    comparisons/.Rproj.user/
    Ignored:    comparisons/code/.DS_Store
    Ignored:    comparisons/code/.ipynb_checkpoints/
    Ignored:    mIF/.DS_Store
    Ignored:    mIF/Analysis/.Rproj.user/
    Ignored:    mIF/Analysis/analysis/.DS_Store
    Ignored:    mIF/Processing/.DS_Store

Untracked files:
    Untracked:  IMC/Analysis/analysis/05_predict_celltypes.Rmd

Unstaged changes:
    Modified:   IMC/Analysis/analysis/01_read_data.Rmd
    Modified:   IMC/Analysis/analysis/03_quality_control.Rmd
    Modified:   IMC/Analysis/analysis/04_sample_correction.Rmd
    Deleted:    IMC/Analysis/analysis/05_labelling.Rmd
    Deleted:    IMC/Analysis/analysis/06_train_classifier.Rmd
    Deleted:    IMC/Analysis/analysis/07_predict_celltypes.Rmd
    Modified:   IMC/Analysis/analysis/_site.yml

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<p>Here, we will use the previously trained random forest classifier to
predict all unlabelled cells.</p>
<div id="load-data" class="section level2">
<h2>Load data</h2>
<p>First, we will read in the SingleCellExperiment object and load all
libraries.</p>
<pre class="r"><code>library(caret)
library(scater)
library(tidyverse)
library(dittoSeq)
library(viridis)
library(doParallel)

sce &lt;- readRDS(&quot;/Volumes/immucan_volume/processed_data/Panel_1/2022_WORKFLOW/IMC/Rout/sce.rds&quot;)
images &lt;- readRDS(&quot;/Volumes/immucan_volume/processed_data/Panel_1/2022_WORKFLOW/IMC/Rout/images.rds&quot;)
masks &lt;- readRDS(&quot;/Volumes/immucan_volume/processed_data/Panel_1/2022_WORKFLOW/IMC/Rout/masks.rds&quot;)</code></pre>
</div>
<div id="prediction-based-on-external-pre-trained-classifier"
class="section level2">
<h2>Prediction based on external pre-trained classifier</h2>
<p>Here, we will use a pre-trained classifier to predict cell types. We
will first adjust the data to match to the classifier</p>
<pre class="r"><code>rffit &lt;- readRDS(&quot;/Volumes/immucan_volume/processed_data/Panel_1/cell_classification/rf_images_DCfix.rds&quot;)

sce$Indication &lt;- sce$indication

sce$Indication[sce$Indication == &quot;BCC&quot;] &lt;- &quot;BREAS&quot;
sce$Indication[sce$Indication == &quot;CRC&quot;] &lt;- &quot;GI&quot;
sce$Indication[sce$Indication == &quot;NSCLC&quot;] &lt;- &quot;GU&quot;
sce$Indication[sce$Indication == &quot;RCC&quot;] &lt;- &quot;HN&quot;
sce$Indication[sce$Indication == &quot;SCCHN&quot;] &lt;- &quot;THOR&quot;</code></pre>
<div id="predict-cells" class="section level3">
<h3>Predict cells</h3>
<p>We will now predict all cells in the dataset</p>
<pre class="r"><code>library(ggridges)

# Add dummy variables
cur_mat &lt;- t(assay(sce, &quot;exprs&quot;)[!grepl(&quot;DNA|Histone&quot;, rownames(sce)),])

dummies &lt;- dummyVars(sample_id ~ Indication, data = colData(sce))
all_dummies &lt;- predict(dummies, newdata = colData(sce))

cur_mat &lt;- cbind(cur_mat, all_dummies)

cell_labels.class &lt;- as.character(predict.train(rffit, 
              newdata = cur_mat, 
              type = &quot;raw&quot;))
names(cell_labels.class) &lt;- rownames(cur_mat)
cell_labels.prob &lt;- predict.train(rffit, 
              newdata = cur_mat, 
              type = &quot;prob&quot;)

cell_labels.class[cell_labels.class == &quot;plasma&quot;] &lt;- &quot;Plasma&quot;
cell_labels.class[cell_labels.class == &quot;B&quot;] &lt;- &quot;Bcell&quot;
cell_labels.class[cell_labels.class == &quot;BnT&quot;] &lt;- &quot;BnTcell&quot;

colnames(cell_labels.prob) &lt;- c(&quot;Bcell&quot;, &quot;BnTcell&quot;, &quot;CD4&quot;, &quot;CD8&quot;, &quot;DC&quot;, 
                                &quot;HLADR&quot;, &quot;MacCD163&quot;, &quot;Mural&quot;, 
                                &quot;Neutrophil&quot;, &quot;NK&quot;, &quot;pDC&quot;, &quot;Plasma&quot;, &quot;Treg&quot;, &quot;Tumor&quot;)

# Distribution of maximum probabilities
tibble(max_prob = rowMax(as.matrix(cell_labels.prob)),
       type = cell_labels.class) %&gt;%
    ggplot() +
        geom_density_ridges(aes(x = max_prob, y = type, fill = type)) +
        scale_fill_manual(values = metadata(sce)$color_vectors$celltype) +
        theme_classic(base_size = 15) +
        xlab(&quot;Maximum probability&quot;) +
        ylab(&quot;Cell type&quot;) + 
        xlim(c(0,1.2))</code></pre>
<pre><code>Picking joint bandwidth of 0.03</code></pre>
<p><img src="figure/05_predict_celltypes.Rmd/predict-cells-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cell_labels.class[rowMax(as.matrix(cell_labels.prob)) &lt; 0.3] &lt;- &quot;undefined&quot;</code></pre>
<p>Store predictions in SCE object. We will not overwrite the labels of
the already labelled cells.</p>
<pre class="r"><code>sce$celltype &lt;- cell_labels.class 
sce$probabilities &lt;- DataFrame(cell_labels.prob)</code></pre>
</div>
<div id="visualization-of-reduced-dimensions" class="section level3">
<h3>Visualization of reduced dimensions</h3>
<p>First, we will plot the class labels on the batch corrected
tSNEs.</p>
<pre class="r"><code>library(dittoSeq)

dittoDimPlot(sce, reduction.use = &quot;UMAP_fastMNN&quot;, var = &quot;celltype&quot;, size = 0.2) +
  scale_color_manual(values = metadata(sce)$color_vectors$celltype)</code></pre>
<p><img src="figure/05_predict_celltypes.Rmd/red-dim-visualization-classlabels-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dittoDimPlot(sce, reduction.use = &quot;UMAP_fastMNN&quot;, var = &quot;indication&quot;, size = 0.2) +
  scale_color_manual(values = metadata(sce)$color_vectors$Indication)</code></pre>
<p><img src="figure/05_predict_celltypes.Rmd/red-dim-visualization-classlabels-2-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="visualization-of-marker-expression" class="section level3">
<h3>Visualization of marker expression</h3>
<p>Finally, we will visualize the marker expression per cell type using
the classified cells.</p>
<pre class="r"><code>agr_sce &lt;- aggregateAcrossCells(sce, 
                                ids = colData(sce)[,c(&quot;sample_id&quot;, &quot;celltype&quot;)], 
                                statistics = &quot;mean&quot;)
assay(agr_sce, &quot;exprs&quot;) &lt;- asinh(counts(agr_sce))
colnames(agr_sce) &lt;- paste0(agr_sce$sample_id, &quot;_&quot;, 
                            agr_sce$celltype)

# Define markers that were used for gating
cur_markers &lt;- c(&quot;Ecad&quot;, &quot;CarbonicAnhydrase&quot;, &quot;Ki67&quot;, &quot;CD14&quot;, &quot;HLADR&quot;, 
                 &quot;CD11c&quot;, &quot;CD163&quot;, &quot;CD303&quot;, &quot;CD68&quot;, &quot;CD15&quot;, &quot;MPO&quot;, &quot;CD38&quot;,
                 &quot;SMA&quot;, &quot;PDGFRb&quot;, &quot;CD20&quot;, &quot;CD7&quot;, &quot;CD3&quot;, &quot;CD8a&quot;, &quot;CD4&quot;, &quot;GrzB&quot;, &quot;FOXP3&quot;, &quot;PD1&quot;)

# Non-scaled
dittoHeatmap(agr_sce[cur_markers,], assay = &quot;exprs&quot;,
            annot.by = c(&quot;patient_id&quot;, &quot;celltype&quot;, &quot;indication&quot;), 
            cluster_rows = FALSE,
            scale = &quot;none&quot;, heatmap.colors = viridis(100), order.by = &quot;celltype&quot;,
            annotation_colors = list(patient_id = metadata(sce)$color_vectors$patient_id,
                                     celltype = metadata(sce)$color_vectors$celltype,
                                     indication = metadata(sce)$color_vectors$Indication))</code></pre>
<p><img src="figure/05_predict_celltypes.Rmd/heatmap-visualization-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Centered and scaled
dittoHeatmap(agr_sce[cur_markers,], assay = &quot;exprs&quot;,
            annot.by = c(&quot;patient_id&quot;, &quot;celltype&quot;, &quot;indication&quot;), 
            cluster_rows = FALSE, order.by = &quot;celltype&quot;,
            annotation_colors = list(patient_id = metadata(sce)$color_vectors$patient_id,
                                     celltype = metadata(sce)$color_vectors$celltype,
                                     indication = metadata(sce)$color_vectors$Indication),
            heatmap.colors = colorRampPalette(c(&quot;dark blue&quot;, &quot;white&quot;, &quot;dark red&quot;))(100),
            breaks = seq(-3, 3, length.out = 101))</code></pre>
<p><img src="figure/05_predict_celltypes.Rmd/heatmap-visualization-2-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="write-out-example-images" class="section level3">
<h3>Write out example images</h3>
<p>Here, we will outline the cells of each cell type onto of composite
images.</p>
<pre class="r"><code>library(SingleCellExperiment)
library(cytomapper)

cell_types &lt;- names(metadata(sce)$colour_vectors$cell_types)
cell_types &lt;- cell_types[cell_types != &quot;unlabelled&quot;]

markers &lt;- vector(mode = &quot;list&quot;, length = length(cell_types))
names(markers) &lt;- cell_types

markers[[&quot;Bcell&quot;]] &lt;- c(&quot;CD20&quot;, &quot;CD3&quot;)
markers[[&quot;Plasma&quot;]] &lt;- c(&quot;CD38&quot;)
markers[[&quot;CD4&quot;]] &lt;- c(&quot;CD4&quot;, &quot;CD8a&quot;, &quot;CD3&quot;)
markers[[&quot;CD8&quot;]] &lt;- c(&quot;CD4&quot;, &quot;CD8a&quot;, &quot;CD3&quot;)
markers[[&quot;Treg&quot;]] &lt;- c(&quot;CD4&quot;, &quot;CD8a&quot;, &quot;FOXP3&quot;)
markers[[&quot;BnTcell&quot;]] &lt;- c(&quot;CD20&quot;, &quot;CD3&quot;)
markers[[&quot;DC&quot;]] &lt;- c(&quot;CD11c&quot;, &quot;CD68&quot;)
markers[[&quot;MacCD163&quot;]] &lt;- c(&quot;CD11c&quot;, &quot;CD163&quot;)
markers[[&quot;HLADR&quot;]] &lt;- c(&quot;HLADR&quot;, &quot;CD11c&quot;)
markers[[&quot;Neutrophil&quot;]] &lt;- c(&quot;CD15&quot;, &quot;MPO&quot;)
markers[[&quot;NK&quot;]] &lt;- c(&quot;CD7&quot;, &quot;CD3&quot;)
markers[[&quot;pDC&quot;]] &lt;- c(&quot;CD303&quot;)
markers[[&quot;Mural&quot;]] &lt;- c(&quot;SMA&quot;, &quot;PDGFRb&quot;)
markers[[&quot;Tumor&quot;]] &lt;- c(&quot;Ecad&quot;, &quot;CarbonicAnhydrase&quot;)
markers[[&quot;undefined&quot;]] &lt;- c(&quot;DNA1&quot;)</code></pre>
<p>Normalize the images.</p>
<pre class="r"><code>images &lt;- normalize(images, separateImages = TRUE)
images &lt;- normalize(images, separateImages = TRUE, 
                        inputRange = c(0, 0.2))</code></pre>
<p>Now, we will loop through the cell-types and indication.</p>
<pre class="r"><code>if (!dir.exists(&quot;/Volumes/immucan_volume/processed_data/Panel_1/2022_WORKFLOW/IMC/Rout/CellTypeValidation/&quot;)) {
  dir.create(&quot;/Volumes/immucan_volume/processed_data/Panel_1/2022_WORKFLOW/IMC/Rout/CellTypeValidation/&quot;)
} 

for(j in unique(sce$celltype)){
    cur_sce &lt;- sce[,sce$celltype == j]
    cur_markers &lt;- markers[[j]]
  
    if(dim(cur_sce)[2] == 0) {
      next(j)
    }
    
    if (length(cur_markers) == 1) {
      cur_col &lt;- &quot;red&quot;
      names(cur_col) &lt;- j
      cur_col &lt;- list(cur_col)
      names(cur_col) &lt;- &quot;celltype&quot;
      plotPixels(image = images, 
                object = cur_sce, 
                mask = masks, 
                cell_id = &quot;ObjectNumber&quot;,
                img_id = &quot;sample_id&quot;, 
                colour_by = cur_markers, 
                outline_by = &quot;celltype&quot;,
                image_title = list(text = names(images),
                                cex = 1), 
                colour = cur_col,
                save_plot = list(filename = paste0(&quot;/Volumes/immucan_volume/processed_data/Panel_1/2022_WORKFLOW/IMC/Rout/CellTypeValidation/&quot;,
                            j, &quot;.png&quot;)))
    } else {
        plotPixels(image = images,
                   object = cur_sce, 
                   mask = masks, 
                   cell_id = &quot;ObjectNumber&quot;,
                   img_id = &quot;sample_id&quot;, 
                   colour_by = cur_markers, 
                   outline_by = &quot;celltype&quot;,
                   image_title = list(text = names(images),
                                  cex = 1),
                   save_plot = list(filename = paste0(&quot;/Volumes/immucan_volume/processed_data/Panel_1/2022_WORKFLOW/IMC/Rout/CellTypeValidation/&quot;,
                            j, &quot;.png&quot;)))
    }
}</code></pre>
</div>
</div>
<div id="define-matched-cell-types" class="section level2">
<h2>Define matched cell types</h2>
<p>To compare to mIF data, we will need to harmonize the cell types.</p>
<pre class="r"><code>matched_celltype &lt;- sce$celltype
matched_celltype[matched_celltype %in% c(&quot;Plasma&quot;, &quot;HLADR&quot;, &quot;Mural&quot;, &quot;pDC&quot;, &quot;NK&quot;, &quot;undefined&quot;)] &lt;- &quot;other&quot;
matched_celltype[matched_celltype %in% c(&quot;Treg&quot;, &quot;CD4&quot;, &quot;CD8&quot;)] &lt;- &quot;Tcell&quot;
matched_celltype[matched_celltype == &quot;TumorKi67&quot;] &lt;- &quot;Tumor&quot;
sce$matched_celltype &lt;- matched_celltype</code></pre>
</div>
<div id="save-rds" class="section level1">
<h1>Save RDS</h1>
<pre class="r"><code>saveRDS(sce, &quot;/Volumes/immucan_volume/processed_data/Panel_1/2022_WORKFLOW/IMC/Rout/sce.rds&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 (2022-10-31)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] cytomapper_1.11.2           EBImage_4.40.0             
 [3] ggridges_0.5.4              doParallel_1.0.17          
 [5] iterators_1.0.14            foreach_1.5.2              
 [7] viridis_0.6.2               viridisLite_0.4.1          
 [9] dittoSeq_1.10.0             forcats_1.0.0              
[11] stringr_1.5.0               dplyr_1.1.0                
[13] purrr_1.0.1                 readr_2.1.3                
[15] tidyr_1.3.0                 tibble_3.1.8               
[17] tidyverse_1.3.2             scater_1.26.1              
[19] scuttle_1.8.4               SingleCellExperiment_1.20.0
[21] SummarizedExperiment_1.28.0 Biobase_2.58.0             
[23] GenomicRanges_1.50.2        GenomeInfoDb_1.34.9        
[25] IRanges_2.32.0              S4Vectors_0.36.1           
[27] BiocGenerics_0.44.0         MatrixGenerics_1.10.0      
[29] matrixStats_0.63.0          caret_6.0-93               
[31] lattice_0.20-45             ggplot2_3.4.0              
[33] workflowr_1.7.0            

loaded via a namespace (and not attached):
  [1] utf8_1.2.3                shinydashboard_0.7.2     
  [3] R.utils_2.12.2            tidyselect_1.2.0         
  [5] htmlwidgets_1.6.1         grid_4.2.2               
  [7] BiocParallel_1.32.5       DropletUtils_1.18.1      
  [9] pROC_1.18.0               munsell_0.5.0            
 [11] ScaledMatrix_1.6.0        codetools_0.2-18         
 [13] future_1.31.0             withr_2.5.0              
 [15] colorspace_2.1-0          highr_0.10               
 [17] knitr_1.42                rstudioapi_0.14          
 [19] listenv_0.9.0             labeling_0.4.2           
 [21] git2r_0.31.0              GenomeInfoDbData_1.2.9   
 [23] farver_2.1.1              pheatmap_1.0.12          
 [25] rhdf5_2.42.0              rprojroot_2.0.3          
 [27] parallelly_1.34.0         vctrs_0.5.2              
 [29] generics_0.1.3            ipred_0.9-13             
 [31] xfun_0.37                 timechange_0.2.0         
 [33] randomForest_4.7-1.1      R6_2.5.1                 
 [35] ggbeeswarm_0.7.1          rsvd_1.0.5               
 [37] locfit_1.5-9.7            bitops_1.0-7             
 [39] rhdf5filters_1.10.0       cachem_1.0.6             
 [41] DelayedArray_0.24.0       assertthat_0.2.1         
 [43] promises_1.2.0.1          scales_1.2.1             
 [45] nnet_7.3-18               googlesheets4_1.0.1      
 [47] beeswarm_0.4.0            gtable_0.3.1             
 [49] beachmat_2.14.0           globals_0.16.2           
 [51] processx_3.8.0            timeDate_4022.108        
 [53] rlang_1.0.6               systemfonts_1.0.4        
 [55] splines_4.2.2             ModelMetrics_1.2.2.2     
 [57] gargle_1.3.0              broom_1.0.3              
 [59] yaml_2.3.7                reshape2_1.4.4           
 [61] abind_1.4-5               modelr_0.1.10            
 [63] backports_1.4.1           httpuv_1.6.8             
 [65] tools_4.2.2               lava_1.7.1               
 [67] SpatialExperiment_1.8.0   ellipsis_0.3.2           
 [69] raster_3.6-14             jquerylib_0.1.4          
 [71] RColorBrewer_1.1-3        Rcpp_1.0.10              
 [73] plyr_1.8.8                sparseMatrixStats_1.10.0 
 [75] zlibbioc_1.44.0           RCurl_1.98-1.10          
 [77] ps_1.7.2                  rpart_4.1.19             
 [79] cowplot_1.1.1             haven_2.5.1              
 [81] ggrepel_0.9.3             fs_1.6.0                 
 [83] magrittr_2.0.3            magick_2.7.3             
 [85] data.table_1.14.6         reprex_2.0.2             
 [87] googledrive_2.0.0         whisker_0.4.1            
 [89] hms_1.1.2                 mime_0.12                
 [91] evaluate_0.20             fftwtools_0.9-11         
 [93] xtable_1.8-4              jpeg_0.1-10              
 [95] readxl_1.4.1              gridExtra_2.3            
 [97] compiler_4.2.2            crayon_1.5.2             
 [99] R.oo_1.25.0               htmltools_0.5.4          
[101] later_1.3.0               tzdb_0.3.0               
[103] tiff_0.1-11               lubridate_1.9.1          
[105] DBI_1.1.3                 dbplyr_2.3.0             
[107] MASS_7.3-58.2             Matrix_1.5-3             
[109] cli_3.6.0                 R.methodsS3_1.8.2        
[111] gower_1.0.1               pkgconfig_2.0.3          
[113] getPass_0.2-2             sp_1.6-0                 
[115] terra_1.7-3               recipes_1.0.4            
[117] xml2_1.3.3                svglite_2.1.1            
[119] vipor_0.4.5               bslib_0.4.2              
[121] hardhat_1.2.0             dqrng_0.3.0              
[123] XVector_0.38.0            prodlim_2019.11.13       
[125] rvest_1.0.3               callr_3.7.3              
[127] digest_0.6.31             rmarkdown_2.20           
[129] cellranger_1.1.0          edgeR_3.40.2             
[131] DelayedMatrixStats_1.20.0 shiny_1.7.4              
[133] rjson_0.2.21              lifecycle_1.0.3          
[135] nlme_3.1-162              jsonlite_1.8.4           
[137] Rhdf5lib_1.20.0           BiocNeighbors_1.16.0     
[139] limma_3.54.1              fansi_1.0.4              
[141] pillar_1.8.1              fastmap_1.1.0            
[143] httr_1.4.4                survival_3.5-0           
[145] glue_1.6.2                svgPanZoom_0.3.4         
[147] png_0.1-8                 nnls_1.4                 
[149] class_7.3-20              stringi_1.7.12           
[151] sass_0.4.5                HDF5Array_1.26.0         
[153] BiocSingular_1.14.0       irlba_2.3.5.1            
[155] future.apply_1.10.0      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
