<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Train random forrest classifier</title>

<script src="site_libs/header-attrs-2.17/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">IMMUcanWorkflow2022</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Train random forrest classifier</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-11-09
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220321code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220321)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220321code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220321)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomImmucanWP7IMMUcanWorkflow2021treeebe28aeb85ab68f57f66ce1618fe58f96a5d35a8targetblankebe28aea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/tree/ebe28aeb85ab68f57f66ce1618fe58f96a5d35a8" target="_blank">ebe28ae</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomImmucanWP7IMMUcanWorkflow2021treeebe28aeb85ab68f57f66ce1618fe58f96a5d35a8targetblankebe28aea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/tree/ebe28aeb85ab68f57f66ce1618fe58f96a5d35a8" target="_blank">ebe28ae</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    Alignment/
    Ignored:    IMC/.DS_Store
    Ignored:    IMC/Analysis/.DS_Store
    Ignored:    IMC/Analysis/.Rproj.user/
    Ignored:    comparisons/.Rproj.user/
    Ignored:    comparisons/code/.ipynb_checkpoints/
    Ignored:    mIF/.DS_Store
    Ignored:    mIF/Analysis/.Rproj.user/

Unstaged changes:
    Modified:   IMC/Analysis/analysis/07_predict_celltypes.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>IMC/Analysis/analysis/06_train_classifier.Rmd</code>) and HTML
(<code>IMC/Analysis/docs/06_train_classifier.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/c28e2279f4ba44f0b4c265f5133bfaf10078372b/IMC/Analysis/analysis/06_train_classifier.Rmd" target="_blank">c28e227</a>
</td>
<td>
nilseling
</td>
<td>
2022-10-21
</td>
<td>
Final classifier
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/e402ead09f7fcf1c50cd3623df965c7764bcd389/IMC/Analysis/analysis/06_train_classifier.Rmd" target="_blank">e402ead</a>
</td>
<td>
nilseling
</td>
<td>
2022-10-13
</td>
<td>
Adjustments for new samples
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ImmucanWP7/IMMUcanWorkflow2021/329cfe69b24cf8a10aa69b2fb4a27b155bfcc091/IMC/Analysis/docs/06_train_classifier.html" target="_blank">329cfe6</a>
</td>
<td>
nilseling
</td>
<td>
2022-08-19
</td>
<td>
Added phenotype key for mIF
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ImmucanWP7/IMMUcanWorkflow2021/4554a259201ef6a52df476c8208dede506b41b04/IMC/Analysis/docs/06_train_classifier.html" target="_blank">4554a25</a>
</td>
<td>
nilseling
</td>
<td>
2022-06-24
</td>
<td>
Recompiled scripts
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ImmucanWP7/IMMUcanWorkflow2021/8d1f8ec230505f81891267366712aeb5847bda1a/IMC/Analysis/docs/06_train_classifier.html" target="_blank">8d1f8ec</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-30
</td>
<td>
Recompiled workflow
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ImmucanWP7/IMMUcanWorkflow2021/f7cc35b2360b342abaa00a75373b5f8a03f47633/IMC/Analysis/docs/06_train_classifier.html" target="_blank">f7cc35b</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-11
</td>
<td>
Recompiled scripts
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/c22d4e1fe74fdd577574bf3c0c514bb5df0a06fa/IMC/Analysis/analysis/06_train_classifier.Rmd" target="_blank">c22d4e1</a>
</td>
<td>
nilseling
</td>
<td>
2022-04-29
</td>
<td>
Added more projects
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>In this script, we will train a random forest classifier.</p>
<div id="read-in-data" class="section level1">
<h1>Read in data</h1>
<p>First, we will read in the SingleCellExperiment object and load all
libraries.</p>
<pre class="r"><code>library(caret)
library(scater)
library(tidyverse)
library(dittoSeq)
library(viridis)
library(doParallel)

sce &lt;- readRDS(&quot;/Volumes/immucan_volume/processed_data/Panel_1/2022_WORKFLOW/IMC/Rout/sce.rds&quot;)</code></pre>
</div>
<div id="random-forrest-training" class="section level1">
<h1>Random Forrest Training</h1>
<p>After quality control, we will now use a random forest classifier to
classify the remaining cells in the dataset. First, we will split the
dataset into labelled and unlabelled data.</p>
<pre class="r"><code>lab_sce &lt;- sce[,sce$cell_labels != &quot;unlabelled&quot;]
unlab_sce &lt;- sce[,sce$cell_labels == &quot;unlabelled&quot;]</code></pre>
<div id="splitting-across-all-samples" class="section level2">
<h2>Splitting across all samples</h2>
<p>We will split the labelled data based on their cell-types.</p>
<div id="train-and-validate-the-classifier" class="section level3">
<h3>Train and validate the classifier</h3>
<p>We will first split the labelled data into training and test
(validation) data at a ratio of 75/25 train/test.</p>
<pre class="r"><code>set.seed(1234)
trainIndex &lt;- createDataPartition(factor(lab_sce$cell_labels), p = 0.75)
train_sce &lt;- lab_sce[,trainIndex$Resample1]
test_sce &lt;- lab_sce[,-trainIndex$Resample1]</code></pre>
<p>Here, we will first use a 5-fold cross-validation by partitioning the
data randomly across the full dataset. We will also use parallel
processing for time reasons. For the <code>randomForrest</code>
classifier, we need to tune the <code>mtry</code> parameter - the number
of variables sampled for each split. We will also add the indication as
dummy variable.</p>
<pre class="r"><code># Define seeds for parallel processing
# Per iteration, we evaluate 10 models while tuning mtry
set.seed(222)
seeds &lt;- vector(mode = &quot;list&quot;, length = 6)
for (i in 1:5) {
  seeds[[i]] &lt;- sample(5000, 5)
}
seeds[[6]] &lt;- sample(5000, 1)

fitControl &lt;- trainControl(method = &quot;cv&quot;,
                           number = 5,
                           seeds = seeds)

cl &lt;- makePSOCKcluster(6, setup_strategy = &quot;sequential&quot;)
registerDoParallel(cl)

# Add dummy variables
cur_mat &lt;- t(assay(train_sce, &quot;exprs&quot;)[!grepl(&quot;DNA|Histone&quot;, rownames(train_sce)),])

dummies &lt;- dummyVars(sample_id ~ indication, data = colData(train_sce))
all_dummies &lt;- predict(dummies, newdata = colData(train_sce))

cur_mat &lt;- cbind(cur_mat, all_dummies)

set.seed(1234)
rffit &lt;- train(x = cur_mat, 
               y = factor(train_sce$cell_labels),
               method = &quot;rf&quot;, ntree = 1000,
               tuneLength = 5,
               trControl = fitControl)
stopCluster(cl)

rffit</code></pre>
<pre><code>Random Forest 

14420 samples
   42 predictor
   14 classes: &#39;Bcell&#39;, &#39;BnTcell&#39;, &#39;CD4&#39;, &#39;CD8&#39;, &#39;DC&#39;, &#39;HLADR&#39;, &#39;MacCD163&#39;, &#39;Mural&#39;, &#39;Neutrophil&#39;, &#39;NK&#39;, &#39;pDC&#39;, &#39;Plasma&#39;, &#39;Treg&#39;, &#39;Tumor&#39; 

No pre-processing
Resampling: Cross-Validated (5 fold) 
Summary of sample sizes: 11536, 11535, 11537, 11535, 11537 
Resampling results across tuning parameters:

  mtry  Accuracy   Kappa    
   2    0.9475725  0.9210397
  12    0.9716366  0.9576972
  22    0.9712897  0.9571797
  32    0.9692785  0.9541925
  42    0.9653951  0.9483966

Accuracy was used to select the optimal model using the largest value.
The final value used for the model was mtry = 12.</code></pre>
<pre class="r"><code>saveRDS(rffit,file = &quot;/Volumes/immucan_volume/processed_data/Panel_1/2022_WORKFLOW/IMC/Rout/rf_classifier.rds&quot;)</code></pre>
<p>We will now have a look at the accuracy measures over iterations. The
only parameter that has been tuned is <code>mtry</code>.</p>
<pre class="r"><code>ggplot(rffit) + 
  geom_errorbar(data = rffit$results,
                aes(ymin = Accuracy - AccuracySD,
                    ymax = Accuracy + AccuracySD),
                width = 0.4)</code></pre>
<p><img src="figure/06_train_classifier.Rmd/accuracy-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-accuracy-1">
Past versions of accuracy-1.png
</button>
</p>
<div id="fig-accuracy-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/4554a259201ef6a52df476c8208dede506b41b04/IMC/Analysis/docs/figure/06_train_classifier.Rmd/accuracy-1.png" target="_blank">4554a25</a>
</td>
<td>
nilseling
</td>
<td>
2022-06-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/f7cc35b2360b342abaa00a75373b5f8a03f47633/IMC/Analysis/docs/figure/06_train_classifier.Rmd/accuracy-1.png" target="_blank">f7cc35b</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We can also compute the confusion matrix:</p>
<pre class="r"><code>confusionMatrix(rffit)</code></pre>
<pre><code>Cross-Validated (5 fold) Confusion Matrix 

(entries are percentual average cell counts across resamples)
 
            Reference
Prediction   Bcell BnTcell  CD4  CD8   DC HLADR MacCD163 Mural Neutrophil   NK
  Bcell        4.2     0.1  0.0  0.0  0.0   0.0      0.0   0.0        0.0  0.0
  BnTcell      0.0     6.7  0.0  0.0  0.0   0.0      0.0   0.0        0.0  0.0
  CD4          0.0     0.0  1.9  0.0  0.0   0.0      0.0   0.0        0.0  0.0
  CD8          0.0     0.0  0.0  2.1  0.0   0.0      0.0   0.0        0.0  0.0
  DC           0.0     0.0  0.0  0.0  1.8   0.1      0.2   0.0        0.0  0.0
  HLADR        0.0     0.0  0.0  0.0  0.3   4.3      0.2   0.0        0.0  0.0
  MacCD163     0.0     0.0  0.0  0.0  0.1   0.0      3.2   0.0        0.0  0.0
  Mural        0.0     0.0  0.0  0.0  0.0   0.1      0.0   6.2        0.0  0.1
  Neutrophil   0.0     0.0  0.0  0.0  0.0   0.2      0.1   0.0        5.4  0.0
  NK           0.0     0.0  0.0  0.0  0.0   0.0      0.0   0.0        0.0  0.1
  pDC          0.0     0.0  0.0  0.0  0.0   0.0      0.0   0.0        0.0  0.0
  Plasma       0.0     0.0  0.0  0.0  0.0   0.0      0.0   0.0        0.0  0.0
  Treg         0.0     0.0  0.0  0.0  0.0   0.0      0.0   0.0        0.0  0.0
  Tumor        0.0     0.0  0.0  0.0  0.2   0.3      0.1   0.1        0.0  0.1
            Reference
Prediction    pDC Plasma Treg Tumor
  Bcell       0.0    0.0  0.0   0.0
  BnTcell     0.0    0.0  0.0   0.0
  CD4         0.0    0.0  0.0   0.0
  CD8         0.0    0.0  0.0   0.0
  DC          0.0    0.0  0.0   0.0
  HLADR       0.0    0.0  0.0   0.1
  MacCD163    0.0    0.0  0.0   0.1
  Mural       0.0    0.0  0.0   0.0
  Neutrophil  0.0    0.0  0.0   0.0
  NK          0.0    0.0  0.0   0.0
  pDC         2.3    0.0  0.0   0.0
  Plasma      0.0    2.6  0.0   0.0
  Treg        0.0    0.0  1.2   0.0
  Tumor       0.0    0.0  0.0  55.1
                            
 Accuracy (average) : 0.9716</code></pre>
<p>We will also look at the variable importance.</p>
<pre class="r"><code>cur_varImp &lt;- varImp(rffit)
plot(cur_varImp)</code></pre>
<p><img src="figure/06_train_classifier.Rmd/variable-importance-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-variable-importance-1">
Past versions of variable-importance-1.png
</button>
</p>
<div id="fig-variable-importance-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/4554a259201ef6a52df476c8208dede506b41b04/IMC/Analysis/docs/figure/06_train_classifier.Rmd/variable-importance-1.png" target="_blank">4554a25</a>
</td>
<td>
nilseling
</td>
<td>
2022-06-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/f7cc35b2360b342abaa00a75373b5f8a03f47633/IMC/Analysis/docs/figure/06_train_classifier.Rmd/variable-importance-1.png" target="_blank">f7cc35b</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Finally, we will validate the model using the test data.</p>
<pre class="r"><code># Add dummy variables
cur_mat &lt;- t(assay(test_sce, &quot;exprs&quot;)[!grepl(&quot;DNA|Histone&quot;, rownames(test_sce)),])

dummies &lt;- dummyVars(sample_id ~ indication, data = colData(test_sce))
all_dummies &lt;- predict(dummies, newdata = colData(test_sce))

cur_mat &lt;- cbind(cur_mat, all_dummies)

cur_pred &lt;- predict(rffit, 
                    newdata = cur_mat)

cm &lt;- confusionMatrix(data = cur_pred, 
                      reference = factor(test_sce$cell_labels), 
                      mode = &quot;everything&quot;)
cm</code></pre>
<pre><code>Confusion Matrix and Statistics

            Reference
Prediction   Bcell BnTcell  CD4  CD8   DC HLADR MacCD163 Mural Neutrophil   NK
  Bcell        201       2    0    0    0     0        0     0          0    0
  BnTcell        1     326    0    0    0     0        0     0          1    0
  CD4            0       0   91    1    0     0        0     0          0    0
  CD8            0       0    1   99    0     0        0     0          0    0
  DC             0       0    0    0   86     5        7     0          1    0
  HLADR          0       0    0    0   13   211        8     0          1    0
  MacCD163       0       0    0    0    8     2      153     1          0    0
  Mural          0       0    0    0    0     3        0   301          1    2
  Neutrophil     0       0    2    3    4     3        5     0        258    2
  NK             0       0    0    0    0     0        0     0          0    8
  pDC            0       0    0    0    0     0        0     0          0    0
  Plasma         0       0    0    0    0     0        0     0          0    1
  Treg           0       0    0    0    0     0        1     0          0    0
  Tumor          0       0    0    0    7    10        6     0          0    3
            Reference
Prediction    pDC Plasma Treg Tumor
  Bcell         0      0    0     0
  BnTcell       0      0    0     0
  CD4           0      0    2     0
  CD8           0      0    0     0
  DC            0      0    0     1
  HLADR         1      0    0     3
  MacCD163      0      0    0     4
  Mural         0      1    0     3
  Neutrophil    1      0    0     0
  NK            0      0    0     0
  pDC         112      0    0     0
  Plasma        0    126    0     0
  Treg          0      0   57     0
  Tumor         0      1    0  2649

Overall Statistics
                                          
               Accuracy : 0.9746          
                 95% CI : (0.9697, 0.9788)
    No Information Rate : 0.5542          
    P-Value [Acc &gt; NIR] : &lt; 2.2e-16       
                                          
                  Kappa : 0.9621          
                                          
 Mcnemar&#39;s Test P-Value : NA              

Statistics by Class:

                     Class: Bcell Class: BnTcell Class: CD4 Class: CD8
Sensitivity               0.99505        0.99390    0.96809    0.96117
Specificity               0.99957        0.99955    0.99936    0.99979
Pos Pred Value            0.99015        0.99390    0.96809    0.99000
Neg Pred Value            0.99978        0.99955    0.99936    0.99915
Precision                 0.99015        0.99390    0.96809    0.99000
Recall                    0.99505        0.99390    0.96809    0.96117
F1                        0.99259        0.99390    0.96809    0.97537
Prevalence                0.04208        0.06833    0.01958    0.02146
Detection Rate            0.04188        0.06792    0.01896    0.02063
Detection Prevalence      0.04229        0.06833    0.01958    0.02083
Balanced Accuracy         0.99731        0.99673    0.98372    0.98048
                     Class: DC Class: HLADR Class: MacCD163 Class: Mural
Sensitivity            0.72881      0.90171         0.85000      0.99669
Specificity            0.99701      0.99431         0.99675      0.99778
Pos Pred Value         0.86000      0.89030         0.91071      0.96785
Neg Pred Value         0.99319      0.99496         0.99417      0.99978
Precision              0.86000      0.89030         0.91071      0.96785
Recall                 0.72881      0.90171         0.85000      0.99669
F1                     0.78899      0.89597         0.87931      0.98206
Prevalence             0.02458      0.04875         0.03750      0.06292
Detection Rate         0.01792      0.04396         0.03188      0.06271
Detection Prevalence   0.02083      0.04938         0.03500      0.06479
Balanced Accuracy      0.86291      0.94801         0.92338      0.99723
                     Class: Neutrophil Class: NK Class: pDC Class: Plasma
Sensitivity                    0.98473  0.500000    0.98246       0.98438
Specificity                    0.99559  1.000000    1.00000       0.99979
Pos Pred Value                 0.92806  1.000000    1.00000       0.99213
Neg Pred Value                 0.99912  0.998331    0.99957       0.99957
Precision                      0.92806  1.000000    1.00000       0.99213
Recall                         0.98473  0.500000    0.98246       0.98438
F1                             0.95556  0.666667    0.99115       0.98824
Prevalence                     0.05458  0.003333    0.02375       0.02667
Detection Rate                 0.05375  0.001667    0.02333       0.02625
Detection Prevalence           0.05792  0.001667    0.02333       0.02646
Balanced Accuracy              0.99016  0.750000    0.99123       0.99208
                     Class: Treg Class: Tumor
Sensitivity              0.96610       0.9959
Specificity              0.99979       0.9874
Pos Pred Value           0.98276       0.9899
Neg Pred Value           0.99958       0.9948
Precision                0.98276       0.9899
Recall                   0.96610       0.9959
F1                       0.97436       0.9929
Prevalence               0.01229       0.5542
Detection Rate           0.01188       0.5519
Detection Prevalence     0.01208       0.5575
Balanced Accuracy        0.98295       0.9916</code></pre>
<pre class="r"><code>data.frame(cm$byClass) %&gt;%
  mutate(class = sub(&quot;Class: &quot;, &quot;&quot;, rownames(cm$byClass))) %&gt;%
  ggplot() + 
  geom_point(aes(1 - Specificity, Sensitivity, 
                 size = Detection.Rate,
                 fill = class),
             shape = 21) + 
  scale_fill_manual(values = metadata(sce)$colour_vectors$cell_type) + 
  theme_bw() + 
  ylab(&quot;Sensitivity (TPR)&quot;) +
  xlab(&quot;1 - Specificity (FPR)&quot;)</code></pre>
<p><img src="figure/06_train_classifier.Rmd/model-testing-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-model-testing-1">
Past versions of model-testing-1.png
</button>
</p>
<div id="fig-model-testing-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/4554a259201ef6a52df476c8208dede506b41b04/IMC/Analysis/docs/figure/06_train_classifier.Rmd/model-testing-1.png" target="_blank">4554a25</a>
</td>
<td>
nilseling
</td>
<td>
2022-06-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/f7cc35b2360b342abaa00a75373b5f8a03f47633/IMC/Analysis/docs/figure/06_train_classifier.Rmd/model-testing-1.png" target="_blank">f7cc35b</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We will also observe the distribution of classification probabilities
per image and class:</p>
<pre class="r"><code>cur_pred &lt;- predict(rffit, 
                    newdata = cur_mat, 
                    type = &quot;prob&quot;)
cur_pred$truth &lt;- factor(test_sce$cell_labels)

cur_pred %&gt;%
  pivot_longer(cols = Bcell:Tumor) %&gt;%
  ggplot() +
  geom_boxplot(aes(x = name, y = value, fill = name), outlier.size = 0.5) +
  facet_wrap(. ~ truth) + 
  scale_fill_manual(values = metadata(sce)$colour_vectors$cell_type) +
  theme(panel.background = element_blank(), 
        axis.text.x = element_text(angle = 45, hjust = 1))</code></pre>
<p><img src="figure/06_train_classifier.Rmd/prediciton-probability-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-prediciton-probability-1">
Past versions of prediciton-probability-1.png
</button>
</p>
<div id="fig-prediciton-probability-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/4554a259201ef6a52df476c8208dede506b41b04/IMC/Analysis/docs/figure/06_train_classifier.Rmd/prediciton-probability-1.png" target="_blank">4554a25</a>
</td>
<td>
nilseling
</td>
<td>
2022-06-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ImmucanWP7/IMMUcanWorkflow2021/blob/f7cc35b2360b342abaa00a75373b5f8a03f47633/IMC/Analysis/docs/figure/06_train_classifier.Rmd/prediciton-probability-1.png" target="_blank">f7cc35b</a>
</td>
<td>
nilseling
</td>
<td>
2022-05-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] doParallel_1.0.17           iterators_1.0.14           
 [3] foreach_1.5.2               viridis_0.6.2              
 [5] viridisLite_0.4.1           dittoSeq_1.10.0            
 [7] forcats_0.5.2               stringr_1.4.1              
 [9] dplyr_1.0.10                purrr_0.3.5                
[11] readr_2.1.3                 tidyr_1.2.1                
[13] tibble_3.1.8                tidyverse_1.3.2            
[15] scater_1.26.0               scuttle_1.8.0              
[17] SingleCellExperiment_1.20.0 SummarizedExperiment_1.28.0
[19] Biobase_2.58.0              GenomicRanges_1.50.0       
[21] GenomeInfoDb_1.34.1         IRanges_2.32.0             
[23] S4Vectors_0.36.0            BiocGenerics_0.44.0        
[25] MatrixGenerics_1.10.0       matrixStats_0.62.0         
[27] caret_6.0-93                lattice_0.20-45            
[29] ggplot2_3.4.0               workflowr_1.7.0            

loaded via a namespace (and not attached):
  [1] readxl_1.4.1              backports_1.4.1          
  [3] plyr_1.8.7                splines_4.2.1            
  [5] BiocParallel_1.32.0       listenv_0.8.0            
  [7] digest_0.6.30             htmltools_0.5.3          
  [9] fansi_1.0.3               magrittr_2.0.3           
 [11] ScaledMatrix_1.6.0        googlesheets4_1.0.1      
 [13] tzdb_0.3.0                recipes_1.0.2            
 [15] globals_0.16.1            modelr_0.1.9             
 [17] gower_1.0.0               hardhat_1.2.0            
 [19] timechange_0.1.1          colorspace_2.0-3         
 [21] rvest_1.0.3               ggrepel_0.9.2            
 [23] haven_2.5.1               xfun_0.34                
 [25] crayon_1.5.2              callr_3.7.3              
 [27] RCurl_1.98-1.9            jsonlite_1.8.3           
 [29] survival_3.4-0            glue_1.6.2               
 [31] gtable_0.3.1              gargle_1.2.1             
 [33] ipred_0.9-13              zlibbioc_1.44.0          
 [35] XVector_0.38.0            DelayedArray_0.24.0      
 [37] BiocSingular_1.14.0       future.apply_1.10.0      
 [39] scales_1.2.1              pheatmap_1.0.12          
 [41] DBI_1.1.3                 Rcpp_1.0.9               
 [43] proxy_0.4-27              rsvd_1.0.5               
 [45] lava_1.7.0                prodlim_2019.11.13       
 [47] httr_1.4.4                RColorBrewer_1.1-3       
 [49] ellipsis_0.3.2            farver_2.1.1             
 [51] pkgconfig_2.0.3           nnet_7.3-18              
 [53] sass_0.4.2                dbplyr_2.2.1             
 [55] utf8_1.2.2                labeling_0.4.2           
 [57] tidyselect_1.2.0          rlang_1.0.6              
 [59] reshape2_1.4.4            later_1.3.0              
 [61] munsell_0.5.0             cellranger_1.1.0         
 [63] tools_4.2.1               cachem_1.0.6             
 [65] cli_3.4.1                 generics_0.1.3           
 [67] ggridges_0.5.4            broom_1.0.1              
 [69] evaluate_0.18             fastmap_1.1.0            
 [71] yaml_2.3.6                ModelMetrics_1.2.2.2     
 [73] processx_3.8.0            knitr_1.40               
 [75] fs_1.5.2                  randomForest_4.7-1.1     
 [77] future_1.29.0             nlme_3.1-160             
 [79] sparseMatrixStats_1.10.0  whisker_0.4              
 [81] xml2_1.3.3                compiler_4.2.1           
 [83] rstudioapi_0.14           beeswarm_0.4.0           
 [85] e1071_1.7-12              reprex_2.0.2             
 [87] bslib_0.4.1               stringi_1.7.8            
 [89] highr_0.9                 ps_1.7.2                 
 [91] Matrix_1.5-1              vctrs_0.5.0              
 [93] pillar_1.8.1              lifecycle_1.0.3          
 [95] jquerylib_0.1.4           BiocNeighbors_1.16.0     
 [97] cowplot_1.1.1             data.table_1.14.4        
 [99] bitops_1.0-7              irlba_2.3.5.1            
[101] httpuv_1.6.6              R6_2.5.1                 
[103] promises_1.2.0.1          gridExtra_2.3            
[105] vipor_0.4.5               parallelly_1.32.1        
[107] codetools_0.2-18          MASS_7.3-58.1            
[109] assertthat_0.2.1          rprojroot_2.0.3          
[111] withr_2.5.0               GenomeInfoDbData_1.2.9   
[113] hms_1.1.2                 grid_4.2.1               
[115] rpart_4.1.19              beachmat_2.14.0          
[117] timeDate_4021.106         class_7.3-20             
[119] rmarkdown_2.17            DelayedMatrixStats_1.20.0
[121] googledrive_2.0.0         git2r_0.30.1             
[123] getPass_0.2-2             pROC_1.18.0              
[125] lubridate_1.9.0           ggbeeswarm_0.6.0         </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
